*** ./smali/com/android/internal/telephony/cdma/CdmaDataConnectionTracker.smali	2015-04-19 13:47:23.765102944 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/cdma/CdmaDataConnectionTracker.smali	2015-04-19 13:48:53.285106287 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "CdmaDataConnectionTracker.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field private static final DATA_CONNECTION_ACTIVE_PH_LINK_DOWN:I = 0x1
  
***************
*** 2243,2248 ****
--- 2251,2291 ----
      return-object v0
  .end method
  
+ .method getActiveApn()Lcom/android/internal/telephony/ApnSetting;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
+ 
+     return-object v0
+ .end method
+ 
+ .method getDataConnectionAsyncChannels()Ljava/util/HashMap;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "()",
+             "Ljava/util/HashMap",
+             "<",
+             "Ljava/lang/Integer;",
+             "Lcom/android/internal/telephony/DataConnectionAc;",
+             ">;"
+         }
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
+ 
+     return-object v0
+ .end method
+ 
  .method public declared-synchronized getState(Ljava/lang/String;)Lcom/android/internal/telephony/DataConnectionTracker$State;
      .locals 1
      .parameter "apnType"
***************
*** 3103,3108 ****
--- 3146,3154 ----
  .method protected onDataSetupComplete(Landroid/os/AsyncResult;)V
      .locals 5
      .parameter "ar"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v1, 0x0
***************
*** 3127,3132 ****
--- 3173,3180 ----
  
      if-eqz v3, :cond_1
  
+     invoke-static {p0}, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker$Injector;->onDataSetupComplete(Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;)V
+ 
      invoke-direct {p0, v1}, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;->notifyDefaultData(Ljava/lang/String;)V
  
      :goto_0
***************
*** 3393,3398 ****
--- 3441,3449 ----
      .locals 4
      .parameter "connId"
      .parameter "ar"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 3435,3440 ****
--- 3486,3493 ----
  
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;->setState(Lcom/android/internal/telephony/DataConnectionTracker$State;)V
  
+     invoke-static {p0}, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker$Injector;->onDisconnectDone(Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;)V
+ 
      iget-boolean v2, p0, Lcom/android/internal/telephony/cdma/CdmaDataConnectionTracker;->mPendingRestartRadio:Z
  
      if-eqz v2, :cond_1
