*** ./smali/android/provider/CallLog$Calls.smali	2015-04-19 13:47:05.717102270 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/provider/CallLog$Calls.smali	2015-04-19 13:48:33.429105545 +0800
***************
*** 74,85 ****
--- 74,95 ----
  
  .field public static final VOICEMAIL_URI:Ljava/lang/String; = "voicemail_uri"
  
+ .field static sExtraCallLogValues:Landroid/content/ContentValues;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  
  # direct methods
  .method static constructor <clinit>()V
      .locals 3
  
      .prologue
+     const/4 v0, 0x0
+ 
+     sput-object v0, Landroid/provider/CallLog$Calls;->sExtraCallLogValues:Landroid/content/ContentValues;
+ 
      const-string v0, "content://call_log/calls"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 137,142 ****
--- 147,155 ----
      .parameter "callType"
      .parameter "start"
      .parameter "duration"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-virtual/range {p1 .. p1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 160,170 ****
  
      :cond_0
      :goto_0
!     new-instance v12, Landroid/content/ContentValues;
  
!     const/4 v2, 0x5
  
!     invoke-direct {v12, v2}, Landroid/content/ContentValues;-><init>(I)V
  
      .local v12, values:Landroid/content/ContentValues;
      const-string v2, "number"
--- 173,187 ----
  
      :cond_0
      :goto_0
!     new-instance v2, Landroid/content/ContentValues;
  
!     const/4 v3, 0x5
  
!     invoke-direct {v2, v3}, Landroid/content/ContentValues;-><init>(I)V
! 
!     invoke-static {v2}, Landroid/provider/CallLog$Injector;->getExtraCallLogValues(Landroid/content/ContentValues;)Landroid/content/ContentValues;
! 
!     move-result-object v12
  
      .local v12, values:Landroid/content/ContentValues;
      const-string v2, "number"
***************
*** 380,386 ****
      move-result-object v11
  
      .local v11, result:Landroid/net/Uri;
!     invoke-static/range {p1 .. p1}, Landroid/provider/CallLog$Calls;->removeExpiredEntries(Landroid/content/Context;)V
  
      return-object v11
  
--- 397,403 ----
      move-result-object v11
  
      .local v11, result:Landroid/net/Uri;
!     invoke-static/range {p1 .. p1}, Landroid/provider/CallLog$Injector;->removeExpiredEntries(Landroid/content/Context;)V
  
      return-object v11
  
***************
*** 602,604 ****
--- 619,634 ----
  
      return-void
  .end method
+ 
+ .method public static setExtraCallLogValues(Landroid/content/ContentValues;)V
+     .locals 0
+     .parameter "extraCallLogValues"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sput-object p0, Landroid/provider/CallLog$Calls;->sExtraCallLogValues:Landroid/content/ContentValues;
+ 
+     return-void
+ .end method
