*** ./smali/android/nfc/NfcManager.smali	2015-04-19 13:46:57.369101958 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/nfc/NfcManager.smali	2015-04-19 13:48:25.893105264 +0800
***************
*** 64,66 ****
--- 64,79 ----
  
      return-object v0
  .end method
+ 
+ .method public setPollDuration(I)I
+     .locals 1
+     .parameter "pollDurationMs"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
