*** ./smali/com/android/internal/telephony/gsm/GsmDataConnectionTracker.smali	2015-04-19 13:47:22.421102893 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/gsm/GsmDataConnectionTracker.smali	2015-04-19 13:48:51.965106238 +0800
***************
*** 8,14 ****
      value = {
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$2;,
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$RecoveryAction;,
!         Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$ApnChangeObserver;
      }
  .end annotation
  
--- 8,15 ----
      value = {
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$2;,
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$RecoveryAction;,
!         Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$ApnChangeObserver;,
!         Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;
      }
  .end annotation
  
***************
*** 3233,3238 ****
--- 3234,3251 ----
      goto :goto_0
  .end method
  
+ .method static getPolicyDataEnabled()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->sPolicyDataEnabled:Z
+ 
+     return v0
+ .end method
+ 
  .method private getPreferredApn()Lcom/android/internal/telephony/ApnSetting;
      .locals 12
  
***************
*** 3409,3414 ****
--- 3422,3430 ----
  .method private isDataAllowed(Lcom/android/internal/telephony/ApnContext;)Z
      .locals 1
      .parameter "apnContext"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-virtual {p1}, Lcom/android/internal/telephony/ApnContext;->isReady()Z
***************
*** 3417,3423 ****
  
      if-eqz v0, :cond_0
  
!     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isDataAllowed()Z
  
      move-result v0
  
--- 3433,3439 ----
  
      if-eqz v0, :cond_0
  
!     invoke-static {p0, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->isDataAllowed(Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;Lcom/android/internal/telephony/ApnContext;)Z
  
      move-result v0
  
***************
*** 6563,6568 ****
--- 6579,6627 ----
  
  
  # virtual methods
+ .method callApnIdToType(I)Ljava/lang/String;
+     .locals 1
+     .parameter "apnId"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callIsDataAllowed()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isDataAllowed()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
+ .method callIsMmsDataEnabled()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isMmsDataEnabled()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method protected cleanUpAllConnections(ZLjava/lang/String;)V
      .locals 4
      .parameter "tearDown"
***************
*** 7512,7517 ****
--- 7571,7579 ----
  
  .method public getAnyDataEnabled()Z
      .locals 5
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v2, 0x0
***************
*** 7521,7526 ****
--- 7583,7594 ----
      monitor-enter v3
  
      :try_start_0
+     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isMmsDataEnabled()Z
+ 
+     move-result v4
+ 
+     if-nez v4, :cond_1
+ 
      iget-boolean v4, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mInternalDataEnabled:Z
  
      if-eqz v4, :cond_0
***************
*** 7648,7653 ****
--- 7716,7733 ----
      goto :goto_0
  .end method
  
+ .method getInternalDataEnabled()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mInternalDataEnabled:Z
+ 
+     return v0
+ .end method
+ 
  .method protected getLinkCapabilities(Ljava/lang/String;)Landroid/net/LinkCapabilities;
      .locals 4
      .parameter "apnType"
***************
*** 7970,7975 ****
--- 8050,8067 ----
      goto :goto_0
  .end method
  
+ .method getUserDataEnabled()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mUserDataEnabled:Z
+ 
+     return v0
+ .end method
+ 
  .method protected gotoIdleAndNotifyDataConnection(Ljava/lang/String;)V
      .locals 2
      .parameter "reason"
***************
*** 9768,9773 ****
--- 9860,9868 ----
  .method protected onDataSetupComplete(Landroid/os/AsyncResult;)V
      .locals 16
      .parameter "ar"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      sget-object v3, Lcom/android/internal/telephony/DataConnection$FailCause;->UNKNOWN:Lcom/android/internal/telephony/DataConnection$FailCause;
***************
*** 9968,9973 ****
--- 10063,10070 ----
      .end local v10           #proxy:Landroid/net/ProxyProperties;
      :cond_4
      :goto_3
+     invoke-static {v2}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->onDataSetupComplete(Lcom/android/internal/telephony/ApnContext;)V
+ 
      invoke-virtual {v2}, Lcom/android/internal/telephony/ApnContext;->getApnType()Ljava/lang/String;
  
      move-result-object v12
***************
*** 10449,10454 ****
--- 10546,10554 ----
      .locals 5
      .parameter "connId"
      .parameter "ar"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v4, 0x0
***************
*** 10492,10497 ****
--- 10592,10599 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/telephony/ApnContext;->setState(Lcom/android/internal/telephony/DataConnectionTracker$State;)V
  
+     invoke-static {v0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->onDisconnectDone(Lcom/android/internal/telephony/ApnContext;)V
+ 
      iget-object v1, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/ApnContext;->getReason()Ljava/lang/String;
***************
*** 10581,10590 ****
--- 10683,10701 ----
      .locals 4
      .parameter "apnId"
      .parameter "enabled"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v1, 0x1
  
+     invoke-static {p0, p2, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->getApnType(Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;II)Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     iput-object v2, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
+ 
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mApnContexts:Ljava/util/concurrent/ConcurrentHashMap;
  
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
