*** ./smali/com/android/internal/telephony/gsm/GsmServiceStateTracker.smali	2015-04-19 13:47:22.365102891 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/gsm/GsmServiceStateTracker.smali	2015-04-19 13:48:51.905106235 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$4;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$4;,
!         Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$Injector;
      }
  .end annotation
  
***************
*** 1623,1628 ****
--- 1624,1632 ----
  
  .method private pollStateDone()V
      .locals 38
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      new-instance v34, Ljava/lang/StringBuilder;
***************
*** 2549,2555 ****
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/gsm/GSMPhone;->setSystemProperty(Ljava/lang/String;Ljava/lang/String;)V
  
!     if-nez v24, :cond_18
  
      const-string v34, "operatorNumeric is null"
  
--- 2553,2563 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/gsm/GSMPhone;->setSystemProperty(Ljava/lang/String;Ljava/lang/String;)V
  
!     invoke-static/range {v24 .. v24}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
! 
!     move-result v34
! 
!     if-eqz v34, :cond_18
  
      const-string v34, "operatorNumeric is null"
  
***************
*** 7973,7978 ****
--- 7981,7989 ----
  
  .method protected updateSpnDisplay()V
      .locals 12
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v11, 0x2
***************
*** 8002,8007 ****
--- 8013,8022 ----
  
      invoke-virtual {v8}, Lcom/android/internal/telephony/IccRecords;->getServiceProviderName()Ljava/lang/String;
  
+     move-result-object v8
+ 
+     invoke-static {p0, v8}, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$Injector;->getSpn(Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v5
  
      .local v5, spn:Ljava/lang/String;
***************
*** 8009,8014 ****
--- 8024,8033 ----
  
      invoke-virtual {v8}, Landroid/telephony/ServiceState;->getOperatorAlphaLong()Ljava/lang/String;
  
+     move-result-object v8
+ 
+     invoke-static {p0, v8}, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$Injector;->getPlmn(Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v1
  
      .local v1, plmn:Ljava/lang/String;
