*** ./smali/com/android/internal/telephony/gsm/SpnOverride.smali	2015-04-19 13:47:22.221102886 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/gsm/SpnOverride.smali	2015-04-19 13:48:51.785106231 +0800
***************
*** 271,273 ****
--- 271,286 ----
  
      return-object v0
  .end method
+ 
+ .method getSpn(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+     .locals 1
+     .parameter "carrier"
+     .parameter "spn"
+ 
+     .prologue
+     invoke-virtual {p0, p1}, Lcom/android/internal/telephony/gsm/SpnOverride;->getSpn(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
