*** ./smali/android/content/pm/PackageParser$PackageLite.smali	2015-04-19 13:46:42.737101411 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageParser$PackageLite.smali	2015-04-19 13:48:11.365104721 +0800
***************
*** 21,31 ****
  
  .field public final verifiers:[Landroid/content/pm/VerifierInfo;
  
  
  # direct methods
! .method public constructor <init>(Ljava/lang/String;ILjava/util/List;)V
      .locals 1
      .parameter "packageName"
      .parameter "installLocation"
      .parameter
      .annotation system Ldalvik/annotation/Signature;
--- 21,34 ----
  
  .field public final verifiers:[Landroid/content/pm/VerifierInfo;
  
+ .field public final versionCode:I
+ 
  
  # direct methods
! .method public constructor <init>(Ljava/lang/String;IILjava/util/List;)V
      .locals 1
      .parameter "packageName"
+     .parameter "versionCode"
      .parameter "installLocation"
      .parameter
      .annotation system Ldalvik/annotation/Signature;
***************
*** 41,60 ****
      .end annotation
  
      .prologue
!     .local p3, verifiers:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/VerifierInfo;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      iput-object p1, p0, Landroid/content/pm/PackageParser$PackageLite;->packageName:Ljava/lang/String;
  
!     iput p2, p0, Landroid/content/pm/PackageParser$PackageLite;->installLocation:I
  
!     invoke-interface {p3}, Ljava/util/List;->size()I
  
      move-result v0
  
      new-array v0, v0, [Landroid/content/pm/VerifierInfo;
  
!     invoke-interface {p3, v0}, Ljava/util/List;->toArray([Ljava/lang/Object;)[Ljava/lang/Object;
  
      move-result-object v0
  
--- 44,65 ----
      .end annotation
  
      .prologue
!     .local p4, verifiers:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/VerifierInfo;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      iput-object p1, p0, Landroid/content/pm/PackageParser$PackageLite;->packageName:Ljava/lang/String;
  
!     iput p2, p0, Landroid/content/pm/PackageParser$PackageLite;->versionCode:I
! 
!     iput p3, p0, Landroid/content/pm/PackageParser$PackageLite;->installLocation:I
  
!     invoke-interface {p4}, Ljava/util/List;->size()I
  
      move-result v0
  
      new-array v0, v0, [Landroid/content/pm/VerifierInfo;
  
!     invoke-interface {p4, v0}, Ljava/util/List;->toArray([Ljava/lang/Object;)[Ljava/lang/Object;
  
      move-result-object v0
  
***************
*** 64,66 ****
--- 69,97 ----
  
      return-void
  .end method
+ 
+ .method public constructor <init>(Ljava/lang/String;ILjava/util/List;)V
+     .locals 1
+     .parameter "packageName"
+     .parameter "installLocation"
+     .parameter
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "(",
+             "Ljava/lang/String;",
+             "I",
+             "Ljava/util/List",
+             "<",
+             "Landroid/content/pm/VerifierInfo;",
+             ">;)V"
+         }
+     .end annotation
+ 
+     .prologue
+     .local p3, verifiers:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/VerifierInfo;>;"
+     const/4 v0, 0x0
+ 
+     invoke-direct {p0, p1, v0, p2, p3}, Landroid/content/pm/PackageParser$PackageLite;-><init>(Ljava/lang/String;IILjava/util/List;)V
+ 
+     return-void
+ .end method
