*** ./smali/android/content/pm/PackageManager.smali	2015-04-19 13:46:43.221101429 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageManager.smali	2015-04-19 13:48:11.849104739 +0800
***************
*** 14,27 ****
--- 14,36 ----
  # static fields
  .field public static final ACTION_CLEAN_EXTERNAL_STORAGE:Ljava/lang/String; = "android.content.pm.CLEAN_EXTERNAL_STORAGE"
  
+ .field public static final COMPONENT_ENABLED_STATE_ACCESS_CONTROL:I = -0x80000000
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field public static final COMPONENT_ENABLED_STATE_DEFAULT:I = 0x0
  
  .field public static final COMPONENT_ENABLED_STATE_DISABLED:I = 0x2
  
  .field public static final COMPONENT_ENABLED_STATE_DISABLED_USER:I = 0x3
  
+ .field public static final COMPONENT_ENABLED_STATE_DISABLE_AUTOSTART:I = 0x40000000
  .field public static final COMPONENT_ENABLED_STATE_ENABLED:I = 0x1
  
+ .field public static final COMPONENT_ENABLED_STATE_SHOW_FLOATING_WINDOW:I = 0x8000000
+ 
  .field public static final DELETE_FAILED_DEVICE_POLICY_MANAGER:I = -0x2
  
  .field public static final DELETE_FAILED_INTERNAL_ERROR:I = -0x1
***************
*** 146,151 ****
--- 155,172 ----
  
  .field public static final GET_URI_PERMISSION_PATTERNS:I = 0x800
  
+ .field public static final HAS_ACTIVITY:I = 0x20000
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
+ .field public static final HAS_ACTIVITY_OR_SERVICES:I = 0x40000
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field public static final INSTALL_ALLOW_TEST:I = 0x4
  
  .field public static final INSTALL_EXTERNAL:I = 0x8
***************
*** 156,161 ****
--- 177,184 ----
  
  .field public static final INSTALL_FAILED_CONTAINER_ERROR:I = -0x12
  
+ .field public static final INSTALL_FAILED_CONTAIN_VIRUS:I = -0x19
+ 
  .field public static final INSTALL_FAILED_CPU_ABI_INCOMPATIBLE:I = -0x10
  
  .field public static final INSTALL_FAILED_DEXOPT:I = -0xb
***************
*** 190,195 ****
--- 213,220 ----
  
  .field public static final INSTALL_FAILED_SHARED_USER_INCOMPATIBLE:I = -0x8
  
+ .field public static final INSTALL_FAILED_SYSTEM_INCOMPATIBLE:I = -0x1a
+ 
  .field public static final INSTALL_FAILED_TEST_ONLY:I = -0xf
  
  .field public static final INSTALL_FAILED_UID_CHANGED:I = -0x18
