*** ./smali/android/content/pm/PackageParser.smali	2015-04-19 13:46:43.521101441 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageParser.smali	2015-04-19 13:48:12.141104750 +0800
***************
*** 21,27 ****
          Landroid/content/pm/PackageParser$ParseComponentArgs;,
          Landroid/content/pm/PackageParser$ParsePackageItemArgs;,
          Landroid/content/pm/PackageParser$SplitPermissionInfo;,
!         Landroid/content/pm/PackageParser$NewPermissionInfo;
      }
  .end annotation
  
--- 21,28 ----
          Landroid/content/pm/PackageParser$ParseComponentArgs;,
          Landroid/content/pm/PackageParser$ParsePackageItemArgs;,
          Landroid/content/pm/PackageParser$SplitPermissionInfo;,
!         Landroid/content/pm/PackageParser$NewPermissionInfo;,
!         Landroid/content/pm/PackageParser$Injector;
      }
  .end annotation
  
***************
*** 6330,6335 ****
--- 6331,6340 ----
      .parameter "outInfo"
      .parameter "outError"
      .parameter "isActivity"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Lorg/xmlpull/v1/XmlPullParserException;,
***************
*** 6358,6366 ****
      move-result v7
  
      .local v7, priority:I
      move-object/from16 v0, p5
  
!     invoke-virtual {v0, v7}, Landroid/content/pm/PackageParser$IntentInfo;->setPriority(I)V
  
      const/4 v13, 0x0
  
--- 6363,6377 ----
      move-result v7
  
      .local v7, priority:I
+     move/from16 v0, p4
+ 
+     invoke-static {v0, v7}, Landroid/content/pm/PackageParser$Injector;->checkPriority(II)I
+ 
+     move-result v13
+ 
      move-object/from16 v0, p5
  
!     invoke-virtual {v0, v13}, Landroid/content/pm/PackageParser$IntentInfo;->setPriority(I)V
  
      const/4 v13, 0x0
  
***************
*** 7033,7038 ****
--- 7044,7053 ----
      .parameter "parser"
      .parameter "flags"
      .parameter "outError"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Lorg/xmlpull/v1/XmlPullParserException;,
***************
*** 7206,7211 ****
--- 7221,7232 ----
  
      invoke-static {v0, v3}, Landroid/content/pm/PackageParser;->validateName(Ljava/lang/String;Z)Ljava/lang/String;
  
+     move-result-object v3
+ 
+     move-object/from16 v0, v34
+ 
+     invoke-static {v0, v3}, Landroid/content/pm/PackageParser$Injector;->filterNameError(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v29
  
      .local v29, nameError:Ljava/lang/String;
***************
*** 9265,9270 ****
--- 9286,9295 ----
      .parameter "attrs"
      .parameter "flags"
      .parameter "outError"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Ljava/io/IOException;,
***************
*** 9360,9365 ****
--- 9385,9394 ----
  
      invoke-static {v4, v9}, Landroid/content/pm/PackageParser;->validateName(Ljava/lang/String;Z)Ljava/lang/String;
  
+     move-result-object v9
+ 
+     invoke-static {v4, v9}, Landroid/content/pm/PackageParser$Injector;->filterNameError(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v3
  
      .local v3, nameError:Ljava/lang/String;
***************
*** 9413,9418 ****
--- 9442,9453 ----
      const/4 v2, -0x1
  
      .local v2, installLocation:I
+     const/4 v11, 0x0
+ 
+     .local v11, versionCode:I
+     const/4 v10, 0x0
+ 
+     .local v10, numFound:I
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 9434,9440 ****
  
      move-result v9
  
!     if-eqz v9, :cond_a
  
      const/4 v9, -0x1
  
--- 9469,9475 ----
  
      move-result v9
  
!     if-eqz v9, :cond_miui_0
  
      const/4 v9, -0x1
  
***************
*** 9442,9447 ****
--- 9477,9508 ----
  
      move-result v2
  
+     add-int/lit8 v10, v10, 0x1
+ 
+     goto :cond_miui_1
+ 
+     :cond_miui_0
+     const-string v9, "versionCode"
+ 
+     invoke-virtual {v0, v9}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v9
+ 
+     if-eqz v9, :cond_miui_1
+ 
+     const/4 v9, -0x1
+ 
+     invoke-interface {p2, v1, v9}, Landroid/util/AttributeSet;->getAttributeIntValue(II)I
+ 
+     move-result v11
+ 
+     add-int/lit8 v10, v10, 0x1
+ 
+     :cond_miui_1
+     const/4 v9, 0x2
+ 
+     if-lt v10, v9, :cond_a
+ 
      .end local v0           #attr:Ljava/lang/String;
      :cond_7
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getDepth()I
***************
*** 9533,9539 ****
  
      move-result-object v10
  
!     invoke-direct {v9, v10, v2, v8}, Landroid/content/pm/PackageParser$PackageLite;-><init>(Ljava/lang/String;ILjava/util/List;)V
  
      goto/16 :goto_0
  .end method
--- 9594,9600 ----
  
      move-result-object v10
  
!     invoke-direct {v9, v10, v11, v2, v8}, Landroid/content/pm/PackageParser$PackageLite;-><init>(Ljava/lang/String;IILjava/util/List;)V
  
      goto/16 :goto_0
  .end method
***************
*** 9864,9869 ****
--- 9925,9934 ----
      .parameter "attrs"
      .parameter "flags"
      .parameter "outError"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Ljava/io/IOException;,
***************
*** 9945,9950 ****
--- 10010,10019 ----
      :cond_5
      invoke-static {v1, v7}, Landroid/content/pm/PackageParser;->validateName(Ljava/lang/String;Z)Ljava/lang/String;
  
+     move-result-object v4
+ 
+     invoke-static {v1, v4}, Landroid/content/pm/PackageParser$Injector;->filterNameError(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v0
  
      .local v0, nameError:Ljava/lang/String;
***************
*** 12909,12914 ****
--- 12978,12997 ----
      .local v13, localCerts:[Ljava/security/cert/Certificate;
      if-nez v13, :cond_5
  
+     move-object/from16 v0, p1
+ 
+     iget-object v0, v0, Landroid/content/pm/PackageParser$Package;->packageName:Ljava/lang/String;
+ 
+     move-object/from16 v18, v0
+ 
+     move-object/from16 v0, v18
+ 
+     invoke-static {v12, v0}, Lmiui/content/pm/ExtraPackageManager;->isTrustedAppEntry(Ljava/util/jar/JarEntry;Ljava/lang/String;)Z
+ 
+     move-result v18
+ 
+     if-nez v18, :cond_3
+ 
      const-string v18, "PackageParser"
  
      new-instance v19, Ljava/lang/StringBuilder;
***************
*** 13445,13450 ****
--- 13528,13536 ----
      .parameter "destCodePath"
      .parameter "metrics"
      .parameter "flags"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v6, 0x1
***************
*** 13594,13600 ****
      .local v25, cookie:I
      if-eqz v25, :cond_4
  
!     new-instance v32, Landroid/content/res/Resources;
  
      const/4 v6, 0x0
  
--- 13680,13686 ----
      .local v25, cookie:I
      if-eqz v25, :cond_4
  
!     new-instance v32, Landroid/content/res/MiuiResources;
  
      const/4 v6, 0x0
  
***************
*** 13602,13608 ****
  
      move-object/from16 v1, p3
  
!     invoke-direct {v0, v5, v1, v6}, Landroid/content/res/Resources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;)V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
  
--- 13688,13694 ----
  
      move-object/from16 v1, p3
  
!     invoke-direct {v0, v5, v1, v6}, Landroid/content/res/MiuiResources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;)V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
  
