*** ./smali/android/content/pm/PackageInfoLite.smali	2015-04-19 13:46:43.437101437 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageInfoLite.smali	2015-04-19 13:48:12.057104747 +0800
***************
*** 28,33 ****
--- 28,35 ----
  
  .field public verifiers:[Landroid/content/pm/VerifierInfo;
  
+ .field public versionCode:I
+ 
  
  # direct methods
  .method static constructor <clinit>()V
***************
*** 44,54 ****
  .end method
  
  .method public constructor <init>()V
!     .locals 0
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
  .end method
  
--- 46,60 ----
  .end method
  
  .method public constructor <init>()V
!     .locals 1
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     const/4 v0, 0x0
+ 
+     iput v0, p0, Landroid/content/pm/PackageInfoLite;->versionCode:I
+ 
      return-void
  .end method
  
***************
*** 91,96 ****
--- 97,108 ----
      iput-object v1, p0, Landroid/content/pm/PackageInfoLite;->verifiers:[Landroid/content/pm/VerifierInfo;
  
      :goto_0
+     invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v1
+ 
+     iput v1, p0, Landroid/content/pm/PackageInfoLite;->versionCode:I
+ 
      return-void
  
      :cond_0
***************
*** 214,219 ****
--- 226,235 ----
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
      :goto_0
+     iget v0, p0, Landroid/content/pm/PackageInfoLite;->versionCode:I
+ 
+     invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
+ 
      return-void
  
      :cond_1
