*** ./smali/android/content/pm/ResolveInfo.smali	2015-04-19 13:46:42.769101413 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/pm/ResolveInfo.smali	2015-04-19 13:48:11.381104722 +0800
***************
*** 9,15 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/content/pm/ResolveInfo$DisplayNameComparator;
      }
  .end annotation
  
--- 9,16 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/content/pm/ResolveInfo$DisplayNameComparator;,
!         Landroid/content/pm/ResolveInfo$Injector;
      }
  .end annotation
  
***************
*** 634,639 ****
--- 635,643 ----
  .method public loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
      .locals 6
      .parameter "pm"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
***************
*** 655,660 ****
--- 659,669 ----
      move-result-object v2
  
      .local v2, dr:Landroid/graphics/drawable/Drawable;
+ 
+     invoke-static {v2}, Lmiui/content/res/IconCustomizer;->generateIconStyleDrawable(Landroid/graphics/drawable/Drawable;)Landroid/graphics/drawable/BitmapDrawable;
+ 
+     move-result-object v2
+ 
      if-eqz v2, :cond_1
  
      .end local v2           #dr:Landroid/graphics/drawable/Drawable;
***************
*** 680,688 ****
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
!     iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
  
!     invoke-virtual {p1, v3, v4, v0}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 689,699 ----
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
!     iget-object v4, v1, Landroid/content/pm/ComponentInfo;->name:Ljava/lang/String;
! 
!     iget v5, p0, Landroid/content/pm/ResolveInfo;->icon:I
  
!     invoke-static {p1, v3, v4, v5, v0}, Landroid/app/MiuiThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
