*** ./smali/com/android/internal/policy/impl/KeyguardUpdateMonitor.smali	2015-04-19 13:47:28.261103111 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/KeyguardUpdateMonitor.smali	2015-04-19 13:48:58.005106463 +0800
***************
*** 10,16 ****
          Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallbackImpl;,
          Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;,
          Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$BatteryStatus;,
!         Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$SimArgs;
      }
  .end annotation
  
--- 10,17 ----
          Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallbackImpl;,
          Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;,
          Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$BatteryStatus;,
!         Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$SimArgs;,
!         Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$Injector;
      }
  .end annotation
  
***************
*** 92,97 ****
--- 93,104 ----
      .end annotation
  .end field
  
+ .field private mSkipSimStateChange:Z
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field private mTelephonyPlmn:Ljava/lang/CharSequence;
  
  .field private mTelephonySpn:Ljava/lang/CharSequence;
***************
*** 109,114 ****
--- 116,123 ----
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     iput-boolean v4, p0, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->mSkipSimStateChange:Z
+ 
      sget-object v2, Lcom/android/internal/telephony/IccCard$State;->READY:Lcom/android/internal/telephony/IccCard$State;
  
      iput-object v2, p0, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->mSimState:Lcom/android/internal/telephony/IccCard$State;
***************
*** 684,689 ****
--- 693,705 ----
      .parameter "simArgs"
  
      .prologue
+     iget-boolean v2, p0, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->mSkipSimStateChange:Z
+ 
+     if-eqz v2, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      iget-object v1, p1, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$SimArgs;->simState:Lcom/android/internal/telephony/IccCard$State;
  
      .local v1, state:Lcom/android/internal/telephony/IccCard$State;
***************
*** 866,871 ****
--- 882,890 ----
  .method private static isPluggedIn(Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$BatteryStatus;)Z
      .locals 3
      .parameter "status"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v0, 0x1
***************
*** 881,886 ****
--- 900,911 ----
      if-ne v1, v2, :cond_1
  
      :cond_0
+     invoke-static {}, Landroid/app/MiuiThemeHelper;->isScreenshotMode()Z
+ 
+     move-result v1
+ 
+     if-nez v1, :cond_1
+ 
      :goto_0
      return v0
  
***************
*** 1321,1326 ****
--- 1346,1362 ----
      goto :goto_0
  .end method
  
+ .method isSimPinSecure()Z
+     .locals 1
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->isSimLocked()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public registerInfoCallback(Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;)V
      .locals 3
      .parameter "callback"
***************
*** 1478,1483 ****
--- 1514,1542 ----
      return-void
  .end method
  
+ .method public setFailedAttempts(I)V
+     .locals 0
+     .parameter "failedAttempts"
+ 
+     .prologue
+     iput p1, p0, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->mFailedAttempts:I
+ 
+     return-void
+ .end method
+ 
+ .method public setSkipSimStateChange(Z)V
+     .locals 0
+     .parameter "skip"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->mSkipSimStateChange:Z
+ 
+     return-void
+ .end method
+ 
  .method public shouldShowBatteryInfo()Z
      .locals 1
  
***************
*** 1499,1504 ****
--- 1558,1569 ----
      if-eqz v0, :cond_1
  
      :cond_0
+     invoke-static {}, Landroid/app/MiuiThemeHelper;->isScreenshotMode()Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_1
+ 
      const/4 v0, 0x1
  
      :goto_0
