*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardView$3.smali	2015-04-19 13:47:27.629103088 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardView$3.smali	2015-04-19 13:48:57.325106438 +0800
***************
*** 7,14 ****
  
  
  # annotations
! .annotation system Ldalvik/annotation/EnclosingClass;
!     value = Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
--- 7,14 ----
  
  
  # annotations
! .annotation system Ldalvik/annotation/EnclosingMethod;
!     value = Lcom/android/internal/policy/impl/LockPatternKeyguardView;->createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
***************
*** 392,398 ****
      move-result v2
  
      .local v2, failedAttemptsBeforeWipe:I
!     const/16 v0, 0xf
  
      .local v0, failedAttemptWarning:I
      if-lez v2, :cond_2
--- 392,398 ----
      move-result v2
  
      .local v2, failedAttemptsBeforeWipe:I
!     const/16 v0, 0x8
  
      .local v0, failedAttemptWarning:I
      if-lez v2, :cond_2
***************
*** 466,472 ****
  
      .local v4, showTimeout:Z
      :goto_3
!     if-eqz v5, :cond_5
  
      iget-object v8, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
--- 466,472 ----
  
      .local v4, showTimeout:Z
      :goto_3
!     #if-eqz v5, :cond_5
  
      iget-object v8, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
***************
*** 507,513 ****
  
      .restart local v4       #showTimeout:Z
      :cond_7
!     const/16 v8, 0x14
  
      if-lt v1, v8, :cond_5
  
--- 507,513 ----
  
      .restart local v4       #showTimeout:Z
      :cond_7
!     const/16 v8, 0x9
  
      if-lt v1, v8, :cond_5
  
