*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardView.smali	2015-04-19 13:47:27.085103068 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardView.smali	2015-04-19 13:48:56.825106419 +0800
***************
*** 10,16 ****
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$FastBitmapDrawable;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;,
!         Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
      }
  .end annotation
  
--- 10,17 ----
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$FastBitmapDrawable;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;,
!         Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;,
!         Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;
      }
  .end annotation
  
***************
*** 70,82 ****
  
  .field private mShowLockBeforeUnlock:Z
  
! .field private mSuppressBiometricUnlock:Z
  
  .field private mTransportControlView:Lcom/android/internal/widget/TransportControlView;
  
  .field private mUnlockScreen:Landroid/view/View;
  
! .field private mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
  .field private final mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
--- 71,83 ----
  
  .field private mShowLockBeforeUnlock:Z
  
! .field protected mSuppressBiometricUnlock:Z
  
  .field private mTransportControlView:Lcom/android/internal/widget/TransportControlView;
  
  .field private mUnlockScreen:Landroid/view/View;
  
! .field protected mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
  .field private final mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
***************
*** 100,106 ****
  .end method
  
  .method public constructor <init>(Landroid/content/Context;Lcom/android/internal/policy/impl/KeyguardViewCallback;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardWindowController;)V
!     .locals 3
      .parameter "context"
      .parameter "callback"
      .parameter "updateMonitor"
--- 101,107 ----
  .end method
  
  .method public constructor <init>(Landroid/content/Context;Lcom/android/internal/policy/impl/KeyguardViewCallback;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardWindowController;)V
!     .locals 4
      .parameter "context"
      .parameter "callback"
      .parameter "updateMonitor"
***************
*** 108,113 ****
--- 109,116 ----
      .parameter "controller"
  
      .prologue
+     const/4 v3, 0x1
+ 
      const/4 v2, 0x0
  
      invoke-direct {p0, p1, p2}, Lcom/android/internal/policy/impl/KeyguardViewBase;-><init>(Landroid/content/Context;Lcom/android/internal/policy/impl/KeyguardViewCallback;)V
***************
*** 116,122 ****
  
      iput-boolean v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
  
!     iput-boolean v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mShowLockBeforeUnlock:Z
  
      new-instance v0, Ljava/lang/Object;
  
--- 119,125 ----
  
      iput-boolean v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
  
!     iput-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mShowLockBeforeUnlock:Z
  
      new-instance v0, Ljava/lang/Object;
  
***************
*** 152,160 ****
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mWidgetCallback:Lcom/android/internal/widget/LockScreenWidgetCallback;
  
!     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;
  
!     invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mKeyguardScreenCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
--- 155,163 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mWidgetCallback:Lcom/android/internal/widget/LockScreenWidgetCallback;
  
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
!     move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mKeyguardScreenCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
***************
*** 228,236 ****
  
      invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->registerInfoCallback(Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;)V
  
!     const/4 v0, 0x1
! 
!     invoke-virtual {p0, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->setFocusableInTouchMode(Z)V
  
      const/high16 v0, 0x4
  
--- 231,237 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->registerInfoCallback(Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;)V
  
!     invoke-virtual {p0, v3}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->setFocusableInTouchMode(Z)V
  
      const/high16 v0, 0x4
  
***************
*** 351,357 ****
      .parameter "x0"
  
      .prologue
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showAlmostAtAccountLoginDialog()V
  
      return-void
  .end method
--- 352,358 ----
      .parameter "x0"
  
      .prologue
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showAlmostAtAccountLoginDialog()V
  
      return-void
  .end method
***************
*** 361,367 ****
      .parameter "x0"
  
      .prologue
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showTimeoutDialog()V
  
      return-void
  .end method
--- 362,368 ----
      .parameter "x0"
  
      .prologue
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showTimeoutDialog()V
  
      return-void
  .end method
***************
*** 518,525 ****
--- 519,531 ----
  
  .method private getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
      .locals 2
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
+     invoke-static {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->updateShowLockBeforeUnlock(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCard$State;
***************
*** 641,661 ****
      throw v3
  
      :sswitch_0
!     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Password:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      goto :goto_0
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :sswitch_1
-     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
- 
-     invoke-virtual {v3}, Lcom/android/internal/widget/LockPatternUtils;->isLockPatternEnabled()Z
- 
-     move-result v3
- 
-     if-eqz v3, :cond_4
- 
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
  
      if-nez v3, :cond_2
--- 647,661 ----
      throw v3
  
      :sswitch_0
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockModeForHighPasswordQuality()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
! 
!     move-result-object v0
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      goto :goto_0
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :sswitch_1
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
  
      if-nez v3, :cond_2
***************
*** 676,681 ****
--- 676,689 ----
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :cond_3
+     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     invoke-virtual {v3}, Lcom/android/internal/widget/LockPatternUtils;->isLockPatternEnabled()Z
+ 
+     move-result v3
+ 
+     if-eqz v3, :cond_4
+ 
      sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Pattern:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
***************
*** 699,704 ****
--- 707,743 ----
      .end sparse-switch
  .end method
  
+ .method private getUnlockModeForHighPasswordQuality()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
+ 
+     if-nez v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/widget/LockPatternUtils;->isPermanentlyLocked()Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_1
+ 
+     :cond_0
+     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Account:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     :goto_0
+     return-object v0
+ 
+     :cond_1
+     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Password:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     goto :goto_0
+ .end method
+ 
  .method private initializeBiometricUnlockView(Landroid/view/View;)V
      .locals 7
      .parameter "view"
***************
*** 751,759 ****
  
      if-eqz v2, :cond_3
  
!     const v2, 0x10202b2
! 
!     invoke-virtual {p1, v2}, Landroid/view/View;->findViewById(I)Landroid/view/View;
  
      move-result-object v0
  
--- 790,796 ----
  
      if-eqz v2, :cond_3
  
!     invoke-virtual {p0, p1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getFaceLockAreaView(Landroid/view/View;)Landroid/view/View;
  
      move-result-object v0
  
***************
*** 1015,1022 ****
--- 1052,1073 ----
  
  .method private maybeStartBiometricUnlock()V
      .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     move-result-object v1
+ 
+     sget-object v2, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->LockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     if-ne v1, v2, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mBiometricUnlock:Lcom/android/internal/policy/impl/BiometricSensorUnlock;
  
      if-eqz v1, :cond_0
***************
*** 1082,1089 ****
--- 1133,1152 ----
  
  .method private recreateLockScreen()V
      .locals 2
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
+     invoke-static {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->needRecreateLockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockScreen:Landroid/view/View;
  
      if-eqz v0, :cond_0
***************
*** 1211,1224 ****
      return-void
  .end method
  
! .method private showAlmostAtAccountLoginDialog()V
      .locals 8
  
      .prologue
      const/16 v2, 0x1e
  
      .local v2, timeoutInSeconds:I
!     const/16 v0, 0xf
  
      .local v0, count:I
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 1274,1287 ----
      return-void
  .end method
  
! .method protected showAlmostAtAccountLoginDialog()V
      .locals 8
  
      .prologue
      const/16 v2, 0x1e
  
      .local v2, timeoutInSeconds:I
!     const/16 v0, 0x4
  
      .local v0, count:I
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 1266,1272 ****
      .local v1, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-direct {p0, v3, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
--- 1329,1335 ----
      .local v1, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-virtual {p0, v3, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
***************
*** 1311,1322 ****
      .local v0, message:Ljava/lang/String;
      const/4 v2, 0x0
  
!     invoke-direct {p0, v2, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
  
! .method private showDialog(Ljava/lang/String;Ljava/lang/String;)V
      .locals 4
      .parameter "title"
      .parameter "message"
--- 1374,1385 ----
      .local v0, message:Ljava/lang/String;
      const/4 v2, 0x0
  
!     invoke-virtual {p0, v2, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
  
! .method protected showDialog(Ljava/lang/String;Ljava/lang/String;)V
      .locals 4
      .parameter "title"
      .parameter "message"
***************
*** 1366,1372 ****
      return-void
  .end method
  
! .method private showTimeoutDialog()V
      .locals 7
  
      .prologue
--- 1429,1435 ----
      return-void
  .end method
  
! .method showTimeoutDialog()V
      .locals 7
  
      .prologue
***************
*** 1433,1439 ****
      .local v0, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-direct {p0, v3, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  
--- 1496,1502 ----
      .local v0, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-virtual {p0, v3, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  
***************
*** 1472,1478 ****
      .local v0, message:Ljava/lang/String;
      const/4 v1, 0x0
  
!     invoke-direct {p0, v1, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
--- 1535,1541 ----
      .local v0, message:Ljava/lang/String;
      const/4 v1, 0x0
  
!     invoke-virtual {p0, v1, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
***************
*** 1529,1534 ****
--- 1592,1600 ----
      .locals 6
      .parameter "mode"
      .parameter "force"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iput-object p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
***************
*** 1705,1710 ****
--- 1771,1778 ----
  
      .restart local v2       #visibleScreen:Landroid/view/View;
      :cond_a
+     invoke-static {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->startBiometricUnlock(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
      return-void
  .end method
  
***************
*** 1788,1793 ****
--- 1856,1915 ----
  
  
  # virtual methods
+ .method callGetInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callGetUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callMaybeStartBiometricUnlock()V
+     .locals 0
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->maybeStartBiometricUnlock()V
+ 
+     return-void
+ .end method
+ 
+ .method callStuckOnLockScreenBecauseSimMissing()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public cleanUp()V
      .locals 2
  
***************
*** 1856,1861 ****
--- 1978,1997 ----
      return-void
  .end method
  
+ .method protected createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;
+ 
+     invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
+     return-object v0
+ .end method
+ 
  .method createLockScreen()Landroid/view/View;
      .locals 6
  
***************
*** 1883,1888 ****
--- 2019,2027 ----
  .method createUnlockScreenFor(Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;)Landroid/view/View;
      .locals 10
      .parameter "unlockMode"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v1, 0x0
***************
*** 1892,1898 ****
  
      if-ne p1, v3, :cond_0
  
!     new-instance v0, Lcom/android/internal/policy/impl/PatternUnlockScreen;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
--- 2031,2037 ----
  
      if-ne p1, v3, :cond_0
  
!     new-instance v0, Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
***************
*** 1911,1926 ****
  
      move-result v6
  
!     invoke-direct/range {v0 .. v6}, Lcom/android/internal/policy/impl/PatternUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;I)V
  
!     .local v0, view:Lcom/android/internal/policy/impl/PatternUnlockScreen;
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
  
!     invoke-virtual {v0, v3}, Lcom/android/internal/policy/impl/PatternUnlockScreen;->setEnableFallback(Z)V
  
      move-object v1, v0
  
!     .end local v0           #view:Lcom/android/internal/policy/impl/PatternUnlockScreen;
      .restart local v1       #unlockView:Landroid/view/View;
      :goto_0
      invoke-direct {p0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->initializeTransportControlView(Landroid/view/View;)V
--- 2050,2065 ----
  
      move-result v6
  
!     invoke-direct/range {v0 .. v6}, Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;I)V
  
!     .local v0, view:Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
  
!     invoke-virtual {v0, v3}, Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;->setEnableFallback(Z)V
  
      move-object v1, v0
  
!     .end local v0           #view:Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;
      .restart local v1       #unlockView:Landroid/view/View;
      :goto_0
      invoke-direct {p0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->initializeTransportControlView(Landroid/view/View;)V
***************
*** 1939,1945 ****
  
      if-ne p1, v3, :cond_1
  
!     new-instance v1, Lcom/android/internal/policy/impl/SimPukUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 2078,2084 ----
  
      if-ne p1, v3, :cond_1
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiSimPUKUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 1952,1958 ****
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v1 .. v6}, Lcom/android/internal/policy/impl/SimPukUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
--- 2091,2097 ----
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v1 .. v6}, Lcom/android/internal/policy/impl/MiuiSimPUKUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
***************
*** 1962,1968 ****
  
      if-ne p1, v3, :cond_2
  
!     new-instance v1, Lcom/android/internal/policy/impl/SimUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 2101,2107 ----
  
      if-ne p1, v3, :cond_2
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiSimPINUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 1975,1981 ****
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v1 .. v6}, Lcom/android/internal/policy/impl/SimUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
--- 2114,2120 ----
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v1 .. v6}, Lcom/android/internal/policy/impl/MiuiSimPINUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
***************
*** 1986,1992 ****
      if-ne p1, v3, :cond_3
  
      :try_start_0
!     new-instance v2, Lcom/android/internal/policy/impl/AccountUnlockScreen;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
--- 2125,2131 ----
      if-ne p1, v3, :cond_3
  
      :try_start_0
!     new-instance v2, Lcom/android/internal/policy/impl/MiuiAccountUnlockScreen;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
***************
*** 1998,2004 ****
  
      iget-object v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v2 .. v7}, Lcom/android/internal/policy/impl/AccountUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
      :try_end_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
  
--- 2137,2145 ----
  
      iget-object v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     iget-boolean v8, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
! 
!     invoke-direct/range {v2 .. v8}, Lcom/android/internal/policy/impl/MiuiAccountUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;Z)V
      :try_end_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
  
***************
*** 2034,2040 ****
  
      if-ne p1, v3, :cond_4
  
!     new-instance v1, Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 2175,2181 ----
  
      if-ne p1, v3, :cond_4
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 2049,2057 ****
  
      move-object v3, v1
  
!     invoke-direct/range {v3 .. v8}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
  
      :cond_4
--- 2190,2206 ----
  
      move-object v3, v1
  
!     invoke-direct/range {v3 .. v8}, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
+     move-object v3, v1
+ 
+     check-cast v3, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;
+ 
+     iget-boolean v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
+ 
+     invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;->setEnableFallback(Z)V
+ 
      goto :goto_0
  
      :cond_4
***************
*** 2128,2133 ****
--- 2277,2384 ----
      return v1
  .end method
  
+ .method getConfiguration()Landroid/content/res/Configuration;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mConfiguration:Landroid/content/res/Configuration;
+ 
+     return-object v0
+ .end method
+ 
+ .method protected getFaceLockAreaView(Landroid/view/View;)Landroid/view/View;
+     .locals 1
+     .parameter "view"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     #todo: add alias name to miui res
+     const v0, 0x10202b2
+ 
+     invoke-virtual {p1, v0}, Landroid/view/View;->findViewById(I)Landroid/view/View;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method getLockPatternUtils()Lcom/android/internal/widget/LockPatternUtils;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     return-object v0
+ .end method
+ 
+ .method getLockScreen()Landroid/view/View;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockScreen:Landroid/view/View;
+ 
+     return-object v0
+ .end method
+ 
+ .method getMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     return-object v0
+ .end method
+ 
+ .method getScreenOn()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mScreenOn:Z
+ 
+     return v0
+ .end method
+ 
+ .method getShowLockBeforeUnlock()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mShowLockBeforeUnlock:Z
+ 
+     return v0
+ .end method
+ 
+ .method getUpdateMonitor()Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     return-object v0
+ .end method
+ 
  .method protected onConfigurationChanged(Landroid/content/res/Configuration;)V
      .locals 2
      .parameter "newConfig"
***************
*** 2377,2382 ****
--- 2628,2672 ----
  
      return-void
  .end method
+ 
+ .method setMode(Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     return-void
+ .end method
+ 
+ .method setScreenOn(Z)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mScreenOn:Z
+ 
+     return-void
+ .end method
+ 
+ .method setShowLockBeforeUnlock(Z)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mShowLockBeforeUnlock:Z
+ 
+     return-void
+ .end method
  
  .method public show()V
      .locals 1
