*** ./smali/com/android/internal/policy/impl/PhoneWindow.smali	2015-04-19 13:47:27.109103068 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneWindow.smali	2015-04-19 13:48:56.857106420 +0800
***************
*** 16,22 ****
          Lcom/android/internal/policy/impl/PhoneWindow$DecorView;,
          Lcom/android/internal/policy/impl/PhoneWindow$ActionMenuPresenterCallback;,
          Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;,
!         Lcom/android/internal/policy/impl/PhoneWindow$WindowManagerHolder;
      }
  .end annotation
  
--- 16,23 ----
          Lcom/android/internal/policy/impl/PhoneWindow$DecorView;,
          Lcom/android/internal/policy/impl/PhoneWindow$ActionMenuPresenterCallback;,
          Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;,
!         Lcom/android/internal/policy/impl/PhoneWindow$WindowManagerHolder;,
!         Lcom/android/internal/policy/impl/PhoneWindow$Injector;
      }
  .end annotation
  
***************
*** 1126,1131 ****
--- 1127,1136 ----
      invoke-virtual {v6}, Lcom/android/internal/widget/ActionBarView;->initIndeterminateProgress()V
  
      :cond_7
+     iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindow;->mActionBar:Lcom/android/internal/widget/ActionBarView;
+ 
+     invoke-static {p0, v6}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->handleStartingWindow(Lcom/android/internal/policy/impl/PhoneWindow;Lcom/android/internal/widget/ActionBarView;)V
+ 
      const/4 v2, 0x0
  
      .local v2, splitActionBar:Z
***************
*** 1296,1302 ****
  
      const/4 v3, 0x0
  
!     invoke-static {v2, v3}, Landroid/graphics/drawable/Drawable;->createFromStream(Ljava/io/InputStream;Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
--- 1301,1307 ----
  
      const/4 v3, 0x0
  
!     invoke-static {v3, v3, v2, v3, v3}, Landroid/graphics/drawable/Drawable;->createFromResourceStreamAndCloseStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
***************
*** 1338,1343 ****
--- 1343,1351 ----
      .locals 18
      .parameter "st"
      .parameter "event"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      move-object/from16 v0, p1
***************
*** 1698,1703 ****
--- 1706,1717 ----
  
      iput v4, v2, Landroid/view/WindowManager$LayoutParams;->windowAnimations:I
  
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, v17
+ 
+     invoke-static {v0, v1, v2}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->handleAppLayoutParams(Lcom/android/internal/policy/impl/PhoneWindow;Landroid/view/WindowManager;Landroid/view/WindowManager$LayoutParams;)V
+ 
      move-object/from16 v0, p1
  
      iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->decorView:Lcom/android/internal/policy/impl/PhoneWindow$DecorView;
***************
*** 1761,1767 ****
  
      iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->createdPanelView:Landroid/view/View;
  
!     if-eqz v4, :cond_a
  
      move-object/from16 v0, p1
  
--- 1775,1781 ----
  
      iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->createdPanelView:Landroid/view/View;
  
!     if-eqz v4, :cond_miui_0
  
      move-object/from16 v0, p1
  
***************
*** 1793,1798 ****
--- 1807,1839 ----
      iput v4, v2, Landroid/view/WindowManager$LayoutParams;->gravity:I
  
      goto :goto_6
+ 
+     :cond_miui_0
+     move-object/from16 v0, p1
+ 
+     iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->decorView:Lcom/android/internal/policy/impl/PhoneWindow$DecorView;
+ 
+     invoke-virtual {v4}, Lcom/android/internal/policy/impl/PhoneWindow$DecorView;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
+ 
+     move-result-object v2
+ 
+     .restart local v2       #lp:Landroid/view/ViewGroup$LayoutParams;
+     if-eqz v2, :cond_miui_1
+ 
+     iget v3, v2, Landroid/view/ViewGroup$LayoutParams;->width:I
+ 
+     :cond_miui_1
+     move-object/from16 v0, p1
+ 
+     iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->decorView:Lcom/android/internal/policy/impl/PhoneWindow$DecorView;
+ 
+     invoke-virtual {v4}, Lcom/android/internal/policy/impl/PhoneWindow$DecorView;->getChildCount()I
+ 
+     move-result v4
+ 
+     if-nez v4, :cond_a
+ 
+     goto :goto_0
  .end method
  
  .method private openPanelsAfterRestore()V
***************
*** 2915,2920 ****
--- 2956,2964 ----
  .method protected generateLayout(Lcom/android/internal/policy/impl/PhoneWindow$DecorView;)Landroid/view/ViewGroup;
      .locals 22
      .parameter "decor"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getWindowStyle()Landroid/content/res/TypedArray;
***************
*** 2959,2965 ****
  
      if-eqz v19, :cond_19
  
!     const/16 v19, -0x2
  
      const/16 v20, -0x2
  
--- 3003,3015 ----
  
      if-eqz v19, :cond_19
  
!     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getContext()Landroid/content/Context;
! 
!     move-result-object v19
! 
!     invoke-static/range {v19 .. v19}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->getFloatingWindowWidth(Landroid/content/Context;)I
! 
!     move-result v19
  
      const/16 v20, -0x2
  
***************
*** 3934,3940 ****
  
      move/from16 v1, v19
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/PhoneWindow;->clearFlags(I)V
  
      goto/16 :goto_6
  
--- 3984,3990 ----
  
      move/from16 v1, v19
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/PhoneWindow;->addFlags(I)V
  
      goto/16 :goto_6
  
***************
*** 4085,4098 ****
  
      if-eqz v19, :cond_26
  
!     const v10, 0x1090098
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
  
      .end local v10           #layoutResource:I
      :cond_26
!     const v10, 0x1090097
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
--- 4135,4160 ----
  
      if-eqz v19, :cond_26
  
!     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getContext()Landroid/content/Context;
! 
!     move-result-object v19
! 
!     invoke-static/range {v19 .. v19}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->getActionBarOverlayResourceId(Landroid/content/Context;)I
! 
!     move-result v10
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
  
      .end local v10           #layoutResource:I
      :cond_26
!     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getContext()Landroid/content/Context;
! 
!     move-result-object v19
! 
!     invoke-static/range {v19 .. v19}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->getActionBarResourceId(Landroid/content/Context;)I
! 
!     move-result v10
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
***************
*** 4306,4311 ****
--- 4368,4385 ----
      return-object v4
  .end method
  
+ .method getActionBarView()Lcom/android/internal/widget/ActionBarView;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindow;->mActionBar:Lcom/android/internal/widget/ActionBarView;
+ 
+     return-object v0
+ .end method
+ 
  .method getAudioManager()Landroid/media/AudioManager;
      .locals 2
  
