*** ./smali/com/android/internal/policy/impl/PhoneWindowManager$17.smali	2015-04-19 13:47:28.385103116 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneWindowManager$17.smali	2015-04-19 13:48:58.129106468 +0800
***************
*** 42,51 ****
  
  # virtual methods
  .method public run()V
!     .locals 5
  
      .prologue
!     const/4 v4, 0x0
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$17;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
--- 42,54 ----
  
  # virtual methods
  .method public run()V
!     .locals 6
!     .annotation build Landroid/annotation/MiuiHook;
!         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
!     .end annotation
  
      .prologue
!     const/4 v5, 0x0
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$17;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 61,67 ****
  
      iget-object v3, v3, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
  
!     invoke-direct {v2, p0, v3}, Lcom/android/internal/policy/impl/PhoneWindowManager$17$1;-><init>(Lcom/android/internal/policy/impl/PhoneWindowManager$17;Landroid/content/Context;)V
  
      iput-object v2, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mBootMsgDialog:Landroid/app/ProgressDialog;
  
--- 64,72 ----
  
      iget-object v3, v3, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
  
!     const/4 v4, 0x3
! 
!     invoke-direct {v2, p0, v3, v4}, Lcom/android/internal/policy/impl/PhoneWindowManager$17$1;-><init>(Lcom/android/internal/policy/impl/PhoneWindowManager$17;Landroid/content/Context;I)V
  
      iput-object v2, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mBootMsgDialog:Landroid/app/ProgressDialog;
  
***************
*** 77,83 ****
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mBootMsgDialog:Landroid/app/ProgressDialog;
  
!     invoke-virtual {v1, v4}, Landroid/app/ProgressDialog;->setProgressStyle(I)V
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$17;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
--- 82,88 ----
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mBootMsgDialog:Landroid/app/ProgressDialog;
  
!     invoke-virtual {v1, v5}, Landroid/app/ProgressDialog;->setProgressStyle(I)V
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$17;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 154,160 ****
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mBootMsgDialog:Landroid/app/ProgressDialog;
  
!     invoke-virtual {v1, v4}, Landroid/app/ProgressDialog;->setCancelable(Z)V
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$17;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
--- 159,165 ----
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mBootMsgDialog:Landroid/app/ProgressDialog;
  
!     invoke-virtual {v1, v5}, Landroid/app/ProgressDialog;->setCancelable(Z)V
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$17;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
