*** ./smali/com/android/internal/content/PackageMonitor.smali	2015-04-19 13:47:19.705102792 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/content/PackageMonitor.smali	2015-04-19 13:48:47.329106064 +0800
***************
*** 1114,1119 ****
--- 1114,1135 ----
      goto :goto_0
  .end method
  
+ .method public register(Landroid/content/Context;Z)V
+     .locals 1
+     .parameter "context"
+     .parameter "externalStorage"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, p1, v0, p2}, Lcom/android/internal/content/PackageMonitor;->register(Landroid/content/Context;Landroid/os/Looper;Z)V
+ 
+     return-void
+ .end method
+ 
  .method public unregister()V
      .locals 2
  
