*** ./smali/android/preference/Preference.smali	2015-04-19 13:46:40.721101336 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/preference/Preference.smali	2015-04-19 13:48:06.021104522 +0800
***************
*** 87,92 ****
--- 87,98 ----
  
  .field private mPreferenceManager:Landroid/preference/PreferenceManager;
  
+ .field mPreferenceParent:Landroid/preference/PreferenceGroup;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field private mRequiresKey:Z
  
  .field private mSelectable:Z
***************
*** 1562,1567 ****
--- 1568,1575 ----
  
      const/4 v5, 0x0
  
+     invoke-static {p0, p1}, Landroid/preference/Preference$Injector;->onBindView(Landroid/preference/Preference;Landroid/view/View;)V
+ 
      const v7, 0x1020016
  
      invoke-virtual {p1, v7}, Landroid/view/View;->findViewById(I)Landroid/view/View;
***************
*** 2529,2534 ****
--- 2537,2555 ----
      return-void
  .end method
  
+ .method public setParent(Landroid/preference/PreferenceGroup;)V
+     .locals 0
+     .parameter "parent"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-object p1, p0, Landroid/preference/Preference;->mPreferenceParent:Landroid/preference/PreferenceGroup;
+ 
+     return-void
+ .end method
+ 
  .method public setPersistent(Z)V
      .locals 0
      .parameter "persistent"
