*** ./smali/android/preference/PreferenceScreen.smali	2015-04-19 13:46:40.973101345 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/preference/PreferenceScreen.smali	2015-04-19 13:48:08.717104622 +0800
***************
*** 40,45 ****
--- 40,48 ----
  .method private showDialog(Landroid/os/Bundle;)V
      .locals 7
      .parameter "state"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v6, 0x0
***************
*** 93,105 ****
      move-result-object v4
  
      .local v4, title:Ljava/lang/CharSequence;
!     new-instance v2, Landroid/app/Dialog;
  
      invoke-virtual {v1}, Landroid/content/Context;->getThemeResId()I
  
      move-result v5
  
!     invoke-direct {v2, v1, v5}, Landroid/app/Dialog;-><init>(Landroid/content/Context;I)V
  
      iput-object v2, p0, Landroid/preference/PreferenceScreen;->mDialog:Landroid/app/Dialog;
  
--- 96,108 ----
      move-result-object v4
  
      .local v4, title:Ljava/lang/CharSequence;
!     new-instance v2, Landroid/app/PreferenceDialog;
  
      invoke-virtual {v1}, Landroid/content/Context;->getThemeResId()I
  
      move-result v5
  
!     invoke-direct {v2, v1, v5}, Landroid/app/PreferenceDialog;-><init>(Landroid/content/Context;I)V
  
      iput-object v2, p0, Landroid/preference/PreferenceScreen;->mDialog:Landroid/app/Dialog;
  
***************
*** 157,164 ****
--- 160,175 ----
  
      move-result-object v0
  
+     invoke-static {p1, v0}, Landroid/preference/PreferenceScreen$Injector;->addHeaderView(Landroid/widget/ListView;Landroid/widget/ListAdapter;)V
+ 
+     invoke-virtual {p0}, Landroid/preference/PreferenceScreen;->getRootAdapter()Landroid/widget/ListAdapter;
+ 
+     move-result-object v0
+ 
      invoke-virtual {p1, v0}, Landroid/widget/ListView;->setAdapter(Landroid/widget/ListAdapter;)V
  
+     invoke-static {p1}, Landroid/preference/PreferenceScreen$Injector;->bind(Landroid/widget/ListView;)V
+ 
      invoke-virtual {p0}, Landroid/preference/PreferenceScreen;->onAttachedToActivity()V
  
      return-void
