*** ./smali/android/preference/PreferenceActivity$HeaderAdapter.smali	2015-04-19 13:46:41.221101355 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/preference/PreferenceActivity$HeaderAdapter.smali	2015-04-19 13:48:09.733104660 +0800
***************
*** 81,91 ****
  
      if-nez p2, :cond_0
  
!     iget-object v4, p0, Landroid/preference/PreferenceActivity$HeaderAdapter;->mInflater:Landroid/view/LayoutInflater;
  
!     const v5, 0x1090083
  
!     invoke-virtual {v4, v5, p3, v6}, Landroid/view/LayoutInflater;->inflate(ILandroid/view/ViewGroup;Z)Landroid/view/View;
  
      move-result-object v3
  
--- 81,93 ----
  
      if-nez p2, :cond_0
  
!     invoke-virtual {p0}, Landroid/preference/PreferenceActivity$HeaderAdapter;->getContext()Landroid/content/Context;
  
!     move-result-object v4
! 
!     iget-object v5, p0, Landroid/preference/PreferenceActivity$HeaderAdapter;->mInflater:Landroid/view/LayoutInflater;
  
!     invoke-static {v4, p3, v5}, Landroid/preference/PreferenceActivity$Injector;->getItemView(Landroid/content/Context;Landroid/view/ViewGroup;Landroid/view/LayoutInflater;)Landroid/view/View;
  
      move-result-object v3
  
***************
*** 143,148 ****
--- 145,156 ----
  
      invoke-virtual {v4, v5}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     iget-object v4, v1, Landroid/preference/PreferenceActivity$HeaderAdapter$HeaderViewHolder;->icon:Landroid/widget/ImageView;
+ 
+     iget v5, v0, Landroid/preference/PreferenceActivity$Header;->iconRes:I
+ 
+     invoke-static {v4, v5}, Landroid/preference/PreferenceActivity$Injector;->setIconVisible(Landroid/widget/ImageView;I)V
+ 
      iget-object v4, v1, Landroid/preference/PreferenceActivity$HeaderAdapter$HeaderViewHolder;->title:Landroid/widget/TextView;
  
      invoke-virtual {p0}, Landroid/preference/PreferenceActivity$HeaderAdapter;->getContext()Landroid/content/Context;
