*** ./smali/android/media/AudioService.smali	2015-04-19 13:46:48.841101639 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/media/AudioService.smali	2015-04-19 13:48:17.321104944 +0800
***************
*** 24,30 ****
          Landroid/media/AudioService$SoundPoolCallback;,
          Landroid/media/AudioService$SoundPoolListenerThread;,
          Landroid/media/AudioService$SetModeDeathHandler;,
!         Landroid/media/AudioService$ForceControlStreamClient;
      }
  .end annotation
  
--- 24,31 ----
          Landroid/media/AudioService$SoundPoolCallback;,
          Landroid/media/AudioService$SoundPoolListenerThread;,
          Landroid/media/AudioService$SetModeDeathHandler;,
!         Landroid/media/AudioService$ForceControlStreamClient;,
!         Landroid/media/AudioService$Injector;
      }
  .end annotation
  
***************
*** 351,357 ****
  
  .field private mVolumeControlStream:I
  
! .field private mVolumePanel:Landroid/view/VolumePanel;
  
  
  # direct methods
--- 352,358 ----
  
  .field private mVolumeControlStream:I
  
! .field private mVolumePanel:Lmiui/view/VolumePanel;
  
  
  # direct methods
***************
*** 800,810 ****
  
      sput v0, Landroid/media/AudioService;->SOUND_EFFECT_VOLUME_DB:I
  
!     new-instance v0, Landroid/view/VolumePanel;
  
!     invoke-direct {v0, p1, p0}, Landroid/view/VolumePanel;-><init>(Landroid/content/Context;Landroid/media/AudioService;)V
  
!     iput-object v0, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
      iput v11, p0, Landroid/media/AudioService;->mMode:I
  
--- 801,811 ----
  
      sput v0, Landroid/media/AudioService;->SOUND_EFFECT_VOLUME_DB:I
  
!     new-instance v0, Lmiui/view/VolumePanel;
  
!     invoke-direct {v0, p1, p0}, Lmiui/view/VolumePanel;-><init>(Landroid/content/Context;Landroid/media/AudioService;)V
  
!     iput-object v0, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
      iput v11, p0, Landroid/media/AudioService;->mMode:I
  
***************
*** 1976,1984 ****
      invoke-direct {p0, v0, p2}, Landroid/media/AudioService;->sendVolumeUpdateToRemote(II)V
  
      :cond_1
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v2, p1, p3}, Landroid/view/VolumePanel;->postRemoteVolumeChanged(II)V
  
      goto :goto_0
  
--- 1977,1985 ----
      invoke-direct {p0, v0, p2}, Landroid/media/AudioService;->sendVolumeUpdateToRemote(II)V
  
      :cond_1
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v2, p1, p3}, Lmiui/view/VolumePanel;->postRemoteVolumeChanged(II)V
  
      goto :goto_0
  
***************
*** 5635,5643 ****
  
      iput p3, v2, Landroid/media/AudioService$RemotePlaybackState;->mVolume:I
  
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v2}, Landroid/view/VolumePanel;->postHasNewRemotePlaybackInfo()V
  
      :cond_1
      monitor-exit v4
--- 5636,5644 ----
  
      iput p3, v2, Landroid/media/AudioService$RemotePlaybackState;->mVolume:I
  
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v2}, Lmiui/view/VolumePanel;->postHasNewRemotePlaybackInfo()V
  
      :cond_1
      monitor-exit v4
***************
*** 5674,5682 ****
  
      iput p3, v2, Landroid/media/AudioService$RemotePlaybackState;->mVolumeMax:I
  
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v2}, Landroid/view/VolumePanel;->postHasNewRemotePlaybackInfo()V
  
      :cond_2
      monitor-exit v4
--- 5675,5683 ----
  
      iput p3, v2, Landroid/media/AudioService$RemotePlaybackState;->mVolumeMax:I
  
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v2}, Lmiui/view/VolumePanel;->postHasNewRemotePlaybackInfo()V
  
      :cond_2
      monitor-exit v4
***************
*** 5713,5721 ****
  
      iput p3, v2, Landroid/media/AudioService$RemotePlaybackState;->mVolumeHandling:I
  
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v2}, Landroid/view/VolumePanel;->postHasNewRemotePlaybackInfo()V
  
      :cond_3
      monitor-exit v4
--- 5714,5722 ----
  
      iput p3, v2, Landroid/media/AudioService$RemotePlaybackState;->mVolumeHandling:I
  
!     iget-object v2, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v2}, Lmiui/view/VolumePanel;->postHasNewRemotePlaybackInfo()V
  
      :cond_3
      monitor-exit v4
***************
*** 6047,6055 ****
  
      iput-boolean v0, p0, Landroid/media/AudioService;->mHasRemotePlayback:Z
  
!     iget-object v3, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v3, v0}, Landroid/view/VolumePanel;->postRemoteSliderVisibility(Z)V
  
      :cond_2
      monitor-exit v4
--- 6048,6056 ----
  
      iput-boolean v0, p0, Landroid/media/AudioService;->mHasRemotePlayback:Z
  
!     iget-object v3, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v3, v0}, Lmiui/view/VolumePanel;->postRemoteSliderVisibility(Z)V
  
      :cond_2
      monitor-exit v4
***************
*** 7736,7744 ****
      .parameter "flags"
  
      .prologue
!     iget-object v0, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v0, p2}, Landroid/view/VolumePanel;->postMasterMuteChanged(I)V
  
      invoke-direct {p0, p1}, Landroid/media/AudioService;->broadcastMasterMuteStatus(Z)V
  
--- 7737,7745 ----
      .parameter "flags"
  
      .prologue
!     iget-object v0, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v0, p2}, Lmiui/view/VolumePanel;->postMasterMuteChanged(I)V
  
      invoke-direct {p0, p1}, Landroid/media/AudioService;->broadcastMasterMuteStatus(Z)V
  
***************
*** 7752,7760 ****
      .parameter "newVolume"
  
      .prologue
!     iget-object v1, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v1, p1}, Landroid/view/VolumePanel;->postMasterVolumeChanged(I)V
  
      new-instance v0, Landroid/content/Intent;
  
--- 7753,7761 ----
      .parameter "newVolume"
  
      .prologue
!     iget-object v1, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v1, p1}, Lmiui/view/VolumePanel;->postMasterVolumeChanged(I)V
  
      new-instance v0, Landroid/content/Intent;
  
***************
*** 7864,7872 ****
      const/4 p1, 0x5
  
      :cond_0
!     iget-object v1, p0, Landroid/media/AudioService;->mVolumePanel:Landroid/view/VolumePanel;
  
!     invoke-virtual {v1, p1, p4}, Landroid/view/VolumePanel;->postVolumeChanged(II)V
  
      add-int/lit8 v1, p2, 0x5
  
--- 7865,7873 ----
      const/4 p1, 0x5
  
      :cond_0
!     iget-object v1, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
  
!     invoke-virtual {v1, p1, p4}, Lmiui/view/VolumePanel;->postVolumeChanged(II)V
  
      add-int/lit8 v1, p2, 0x5
  
***************
*** 9365,9370 ****
--- 9366,9374 ----
      .parameter "direction"
      .parameter "suggestedStreamType"
      .parameter "flags"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/16 v3, -0xc8
***************
*** 9409,9414 ****
--- 9413,9424 ----
      and-int/lit8 p3, p3, -0x5
  
      :cond_1
+     iget-object v1, p0, Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
+ 
+     invoke-static {v1, p3, p1}, Landroid/media/AudioService$Injector;->adjustDirection(Lmiui/view/VolumePanel;II)I
+ 
+     move-result p1
+ 
      if-ne v0, v3, :cond_3
  
      and-int/lit8 p3, p3, -0x5
