*** ./smali/com/android/internal/widget/ActionBarContainer.smali	2015-04-19 13:47:17.173102697 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/ActionBarContainer.smali	2015-04-19 13:48:44.693105966 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "ActionBarContainer.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Lcom/android/internal/widget/ActionBarContainer$Injector;
+     }
+ .end annotation
+ 
+ 
  # instance fields
  .field private mActionBarView:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 128,135 ****
      goto :goto_0
  .end method
  
  
- # virtual methods
  .method public getTabContainer()Landroid/view/View;
      .locals 1
  
--- 136,191 ----
      goto :goto_0
  .end method
  
+ .method private shouldPutTabsOnTop()Z
+     .locals 2
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     iget-object v1, p0, Lcom/android/internal/widget/ActionBarContainer;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v1}, Lmiui/util/UiUtils;->isV5Ui(Landroid/content/Context;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_1
+ 
+     :cond_0
+     :goto_0
+     return v0
+ 
+     :cond_1
+     iget-object v1, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/widget/ActionBarView;->getDisplayOptions()I
+ 
+     move-result v1
+ 
+     and-int/lit8 v1, v1, 0x2
+ 
+     if-nez v1, :cond_0
+ 
+     const/4 v0, 0x1
+ 
+     goto :goto_0
+ .end method
+ 
+ 
+ .method protected getActionBarView()Lcom/android/internal/widget/ActionBarView;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
+ 
+     return-object v0
+ .end method
  
  .method public getTabContainer()Landroid/view/View;
      .locals 1
  
***************
*** 139,144 ****
--- 195,212 ----
      return-object v0
  .end method
  
+ .method protected isSplit()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/widget/ActionBarContainer;->mIsSplit:Z
+ 
+     return v0
+ .end method
+ 
  .method public onDraw(Landroid/graphics/Canvas;)V
      .locals 1
      .parameter "canvas"
***************
*** 265,270 ****
--- 333,341 ----
      .parameter "t"
      .parameter "r"
      .parameter "b"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-super/range {p0 .. p5}, Landroid/widget/FrameLayout;->onLayout(ZIIII)V
***************
*** 313,327 ****
      move-result v7
  
      .local v7, tabHeight:I
!     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
! 
!     invoke-virtual {v8}, Lcom/android/internal/widget/ActionBarView;->getDisplayOptions()I
  
      move-result v8
  
!     and-int/lit8 v8, v8, 0x2
! 
!     if-nez v8, :cond_7
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarContainer;->getChildCount()I
  
--- 384,394 ----
      move-result v7
  
      .local v7, tabHeight:I
!     invoke-direct {p0}, Lcom/android/internal/widget/ActionBarContainer;->shouldPutTabsOnTop()Z
  
      move-result v8
  
!     if-eqz v8, :cond_7
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarContainer;->getChildCount()I
  
