*** ./smali/com/android/internal/widget/LockPatternUtils.smali	2015-04-19 13:47:16.349102667 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/LockPatternUtils.smali	2015-04-19 13:48:43.933105938 +0800
***************
*** 8,14 ****
  
  .field protected static final DISABLE_LOCKSCREEN_KEY:Ljava/lang/String; = "lockscreen.disabled"
  
! .field public static final FAILED_ATTEMPTS_BEFORE_RESET:I = 0x14
  
  .field public static final FAILED_ATTEMPTS_BEFORE_TIMEOUT:I = 0x5
  
--- 8,14 ----
  
  .field protected static final DISABLE_LOCKSCREEN_KEY:Ljava/lang/String; = "lockscreen.disabled"
  
! .field public static final FAILED_ATTEMPTS_BEFORE_RESET:I = 0x9
  
  .field public static final FAILED_ATTEMPTS_BEFORE_TIMEOUT:I = 0x5
  
***************
*** 84,89 ****
--- 84,115 ----
      return-void
  .end method
  
+ .method public static callPatternToHash(Ljava/util/List;)[B
+     .locals 1
+     .parameter
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "(",
+             "Ljava/util/List",
+             "<",
+             "Lcom/android/internal/widget/LockPatternView$Cell;",
+             ">;)[B"
+         }
+     .end annotation
+ 
+     .prologue
+     .local p0, pattern:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/widget/LockPatternView$Cell;>;"
+     invoke-static {p0}, Lcom/android/internal/widget/LockPatternUtils;->patternToHash(Ljava/util/List;)[B
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
  .method public static computePasswordQuality(Ljava/lang/String;)I
      .locals 5
      .parameter "password"
***************
*** 271,277 ****
      return-object v0
  .end method
  
! .method private getLong(Ljava/lang/String;J)J
      .locals 3
      .parameter "secureSettingKey"
      .parameter "defaultValue"
--- 297,303 ----
      return-object v0
  .end method
  
! .method protected getLong(Ljava/lang/String;J)J
      .locals 3
      .parameter "secureSettingKey"
      .parameter "defaultValue"
***************
*** 312,318 ****
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-direct {p0, v3, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 338,344 ----
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-virtual {p0, v3, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 334,340 ****
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-direct {p0, v3, v1, v2}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v3, "LockPatternUtils"
  
--- 360,366 ----
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-virtual {p0, v3, v1, v2}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v3, "LockPatternUtils"
  
***************
*** 621,627 ****
      goto :goto_0
  .end method
  
! .method private setLong(Ljava/lang/String;J)V
      .locals 4
      .parameter "secureSettingKey"
      .parameter "value"
--- 647,653 ----
      goto :goto_0
  .end method
  
! .method protected setLong(Ljava/lang/String;J)V
      .locals 4
      .parameter "secureSettingKey"
      .parameter "value"
***************
*** 1112,1122 ****
  
      const-string v0, "lockscreen.password_type"
  
!     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v0, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  .end method
--- 1138,1148 ----
  
      const-string v0, "lockscreen.password_type"
  
!     invoke-virtual {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v0, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  .end method
***************
*** 1196,1202 ****
  
      const-wide/32 v3, 0x10000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v2
  
--- 1222,1228 ----
  
      const-wide/32 v3, 0x10000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v2
  
***************
*** 1354,1359 ****
--- 1380,1395 ----
      return-object v0
  .end method
  
+ .method public getKeyguardLockoutAttemptDeadline(I)J
+     .locals 2
+     .parameter "failedAttempts"
+ 
+     .prologue
+     const-wide/16 v0, -0x1
+ 
+     return-wide v0
+ .end method
+ 
  .method public getKeyguardStoredPasswordQuality()I
      .locals 5
  
***************
*** 1362,1368 ****
  
      const-string v1, "lockscreen.password_type"
  
!     invoke-direct {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 1398,1404 ----
  
      const-string v1, "lockscreen.password_type"
  
!     invoke-virtual {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 1375,1381 ****
  
      const-string v1, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 1411,1417 ----
  
      const-string v1, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 1393,1399 ****
  
      const-string v6, "lockscreen.lockoutattemptdeadline"
  
!     invoke-direct {p0, v6, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 1429,1435 ----
  
      const-string v6, "lockscreen.lockoutattemptdeadline"
  
!     invoke-virtual {p0, v6, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 1704,1710 ****
  
      const-string v2, "lock_biometric_weak_flags"
  
!     invoke-direct {p0, v2, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 1740,1746 ----
  
      const-string v2, "lock_biometric_weak_flags"
  
!     invoke-virtual {p0, v2, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 1782,1795 ****
  
      const-string v8, "lockscreen.password_type"
  
!     invoke-direct {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
      .local v3, mode:J
      const-string v8, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 1818,1831 ----
  
      const-string v8, "lockscreen.password_type"
  
!     invoke-virtual {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
      .local v3, mode:J
      const-string v8, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 1896,1902 ****
  
      const-string v3, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
--- 1932,1938 ----
  
      const-string v3, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
***************
*** 1918,1924 ****
  
      const-string v3, "lockscreen.password_type"
  
!     invoke-direct {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
--- 1954,1960 ----
  
      const-string v3, "lockscreen.password_type"
  
!     invoke-virtual {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
***************
*** 1965,1971 ****
  
      const-string v0, "lockscreen.disabled"
  
!     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 2001,2007 ----
  
      const-string v0, "lockscreen.disabled"
  
!     invoke-virtual {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 2366,2374 ****
  .end method
  
  .method public reportSuccessfulPasswordAttempt()V
!     .locals 1
  
      .prologue
      invoke-virtual {p0}, Lcom/android/internal/widget/LockPatternUtils;->getDevicePolicyManager()Landroid/app/admin/DevicePolicyManager;
  
      move-result-object v0
--- 2402,2416 ----
  .end method
  
  .method public reportSuccessfulPasswordAttempt()V
!     .locals 3
  
      .prologue
+     const-string v0, "lockscreen.lockoutattemptdeadline"
+ 
+     const-wide/16 v1, 0x0
+ 
+     invoke-virtual {p0, v0, v1, v2}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
+ 
      invoke-virtual {p0}, Lcom/android/internal/widget/LockPatternUtils;->getDevicePolicyManager()Landroid/app/admin/DevicePolicyManager;
  
      move-result-object v0
***************
*** 2499,2505 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v3, v11, v12}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      if-eqz v21, :cond_5
  
--- 2541,2547 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v3, v11, v12}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      if-eqz v21, :cond_5
  
***************
*** 2749,2755 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v3, v11, v12}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v3, "lockscreen.password_type_alternate"
  
--- 2791,2797 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v3, v11, v12}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v3, "lockscreen.password_type_alternate"
  
***************
*** 2765,2771 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v3, v11, v12}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct/range {p0 .. p0}, Lcom/android/internal/widget/LockPatternUtils;->finishBiometricWeak()V
  
--- 2807,2813 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v3, v11, v12}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct/range {p0 .. p0}, Lcom/android/internal/widget/LockPatternUtils;->finishBiometricWeak()V
  
***************
*** 2976,2982 ****
  
      const-wide/32 v2, 0x10000
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const/high16 v1, 0x1
  
--- 3018,3024 ----
  
      const-wide/32 v2, 0x10000
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const/high16 v1, 0x1
  
***************
*** 3010,3022 ****
  
      const-wide/32 v2, 0x8000
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v1, "lockscreen.password_type_alternate"
  
      const-wide/32 v2, 0x10000
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->finishBiometricWeak()V
  
--- 3052,3064 ----
  
      const-wide/32 v2, 0x8000
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v1, "lockscreen.password_type_alternate"
  
      const-wide/32 v2, 0x10000
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->finishBiometricWeak()V
  
***************
*** 3180,3186 ****
  
      const-wide/16 v5, 0x0
  
!     invoke-direct {p0, v4, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 3222,3228 ----
  
      const-wide/16 v5, 0x0
  
!     invoke-virtual {p0, v4, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 3195,3201 ****
      :goto_0
      const-string v4, "lock_biometric_weak_flags"
  
!     invoke-direct {p0, v4, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
--- 3237,3243 ----
      :goto_0
      const-string v4, "lock_biometric_weak_flags"
  
!     invoke-virtual {p0, v4, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
***************
*** 3236,3241 ****
--- 3278,3293 ----
      throw v0
  .end method
  
+ .method public setKeyguardLockoutAttemptDeadline(I)J
+     .locals 2
+     .parameter "failedAttempts"
+ 
+     .prologue
+     const-wide/16 v0, -0x1
+ 
+     return-wide v0
+ .end method
+ 
  .method public setLockPatternEnabled(Z)V
      .locals 1
      .parameter "enabled"
***************
*** 3260,3266 ****
      const-wide/16 v0, 0x1
  
      :goto_0
!     invoke-direct {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
--- 3312,3318 ----
      const-wide/16 v0, 0x1
  
      :goto_0
!     invoke-virtual {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
***************
*** 3285,3291 ****
      .local v0, deadline:J
      const-string v2, "lockscreen.lockoutattemptdeadline"
  
!     invoke-direct {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-wide v0
  .end method
--- 3337,3343 ----
      .local v0, deadline:J
      const-string v2, "lockscreen.lockoutattemptdeadline"
  
!     invoke-virtual {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-wide v0
  .end method
***************
*** 3404,3410 ****
  
      const-wide/32 v2, 0x10000
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 3456,3462 ----
  
      const-wide/32 v2, 0x10000
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
