*** ./smali/android/content/ContentResolver.smali	2015-04-19 13:46:44.557101479 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/ContentResolver.smali	2015-04-19 13:48:13.321104794 +0800
***************
*** 3805,3811 ****
  .end method
  
  .method public final query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;Landroid/os/CancellationSignal;)Landroid/database/Cursor;
!     .locals 23
      .parameter "uri"
      .parameter "projection"
      .parameter "selection"
--- 3805,3811 ----
  .end method
  
  .method public final query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;Landroid/os/CancellationSignal;)Landroid/database/Cursor;
!     .locals 30
      .parameter "uri"
      .parameter "projection"
      .parameter "selection"
***************
*** 3853,3858 ****
--- 3853,3859 ----
      invoke-virtual {v0, v9}, Landroid/os/CancellationSignal;->setRemote(Landroid/os/ICancellationSignal;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_1
  
      :cond_2
***************
*** 3871,3876 ****
--- 3872,3878 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/DeadObjectException; {:try_start_1 .. :try_end_1} :catch_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_1 .. :try_end_1} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
      move-result-object v19
***************
*** 3914,3919 ****
--- 3916,3922 ----
      invoke-virtual/range {p0 .. p1}, Landroid/content/ContentResolver;->acquireProvider(Landroid/net/Uri;)Landroid/content/IContentProvider;
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_2 .. :try_end_2} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_1
  
      move-result-object v10
***************
*** 3993,3998 ****
--- 3996,4002 ----
      invoke-direct {v0, v1, v2, v4}, Landroid/content/ContentResolver$CursorWrapperInner;-><init>(Landroid/content/ContentResolver;Landroid/database/Cursor;Landroid/content/IContentProvider;)V
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_3 .. :try_end_3} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_3 .. :try_end_3} :catch_1
  
      .local v22, wrapper:Landroid/content/ContentResolver$CursorWrapperInner;
***************
*** 4015,4020 ****
--- 4019,4025 ----
      invoke-virtual/range {p0 .. p1}, Landroid/content/ContentResolver;->acquireProvider(Landroid/net/Uri;)Landroid/content/IContentProvider;
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_4 .. :try_end_4} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_4 .. :try_end_4} :catch_1
  
      move-result-object v4
***************
*** 4043,4048 ****
--- 4048,4092 ----
      goto :goto_2
  
      .end local v18           #e:Landroid/os/RemoteException;
+     :catch_miui_0
+     move-exception v24
+ 
+     .local v24, e:Ljava/lang/IllegalStateException;
+     :try_start_miui_0
+     move-object/from16 v0, p0
+ 
+     iget-object v0, v0, Landroid/content/ContentResolver;->mContext:Landroid/content/Context;
+ 
+     move-object/from16 v23, v0
+ 
+     move-object/from16 v25, p1
+ 
+     move-object/from16 v26, p2
+ 
+     move-object/from16 v27, p3
+ 
+     move-object/from16 v28, p4
+ 
+     move-object/from16 v29, p5
+ 
+     invoke-static/range {v23 .. v29}, Landroid/content/ContentResolver$Injector;->checkCursorLeakException(Landroid/content/Context;Ljava/lang/IllegalStateException;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)Landroid/database/Cursor;
+     :try_end_miui_0
+     .catchall {:try_start_miui_0 .. :try_end_miui_0} :catchall_0
+ 
+     move-result-object v22
+ 
+     if-eqz v3, :cond_miui_0
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-virtual {v0, v3}, Landroid/content/ContentResolver;->releaseUnstableProvider(Landroid/content/IContentProvider;)Z
+ 
+     :cond_miui_0
+     if-eqz v10, :cond_0
+ 
+     goto :goto_2
+ 
+     .end local v24           #e:Ljava/lang/IllegalStateException;
      :catchall_0
      move-exception v4
  
