*** ./smali/android/view/WindowManager$LayoutParams.smali	2015-04-19 13:46:10.269100199 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/view/WindowManager$LayoutParams.smali	2015-04-19 13:47:39.653103537 +0800
***************
*** 276,281 ****
--- 276,283 ----
  
  .field public dimAmount:F
  
+ .field public extraFlags:I
+ 
  .field public flags:I
      .annotation runtime Landroid/view/ViewDebug$ExportedProperty;
          flagMapping = {
***************
*** 966,971 ****
--- 968,979 ----
  
      move-result v0
  
+     iput v0, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+     
+     invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v0
+ 
      iput v0, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
***************
*** 1316,1321 ****
--- 1324,1342 ----
      or-int/lit8 v0, v0, 0x4
  
      :cond_9
+     iget v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     iget v2, p1, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     if-eq v1, v2, :cond_miui_0
+         
+     iget v1, p1, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     iput v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     or-int/lit8 v0, v0, 0x4
+     
+     :cond_miui_0
      iget v1, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      iget v2, p1, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
***************
*** 1929,1934 ****
--- 1950,1967 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     const-string v1, " extfl=#"
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     iget v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-static {v1}, Ljava/lang/Integer;->toHexString(I)Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      iget v1, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      if-eqz v1, :cond_4
***************
*** 2193,2198 ****
--- 2226,2235 ----
      iget v0, p0, Landroid/view/WindowManager$LayoutParams;->flags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
+     
+     iget v0, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
      iget v0, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
