*** ./smali/com/android/server/am/ActivityStack.smali	2015-04-19 13:47:37.053103440 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/am/ActivityStack.smali	2015-04-19 13:49:06.501106780 +0800
***************
*** 5432,5438 ****
      check-cast v3, Lcom/android/server/am/ActivityRecord;
  
      .local v3, r:Lcom/android/server/am/ActivityRecord;
!     iget-boolean v4, v3, Lcom/android/server/am/ActivityRecord;->finishing:Z
  
      if-eqz v4, :cond_1
  
--- 5432,5440 ----
      check-cast v3, Lcom/android/server/am/ActivityRecord;
  
      .local v3, r:Lcom/android/server/am/ActivityRecord;
!     invoke-static {p3, v3}, Lcom/android/server/am/ActivityStack$Injector;->isDestroyHomeReasonAlwaysOrFinishing(Ljava/lang/String;Lcom/android/server/am/ActivityRecord;)Z
! 
!     move-result v4
  
      if-eqz v4, :cond_1
  
***************
*** 6226,6231 ****
--- 6228,6236 ----
      .locals 7
      .parameter "r"
      .parameter "globalChanges"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v3, 0x1
***************
*** 6316,6321 ****
--- 6321,6336 ----
      if-eqz v5, :cond_c
  
      :cond_7
+     iget-object v5, p1, Lcom/android/server/am/ActivityRecord;->info:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v5, v5, Landroid/content/pm/ActivityInfo;->packageName:Ljava/lang/String;
+ 
+     invoke-static {v5, v0, v1}, Landroid/app/MiuiThemeHelper;->needRestartActivity(Ljava/lang/String;ILandroid/content/res/Configuration;)Z
+ 
+     move-result v5
+ 
+     if-nez v5, :cond_0
+ 
      iget v5, p1, Lcom/android/server/am/ActivityRecord;->configChangeFlags:I
  
      or-int/2addr v5, v0
***************
*** 12276,12281 ****
--- 12291,12309 ----
      move-result-object v12
  
      .local v12, aInfo:Landroid/content/pm/ActivityInfo;
+     move-object/from16 v0, p0
+ 
+     invoke-direct {v0, v12}, Lcom/android/server/am/ActivityStack;->checkRunningCompatibility(Landroid/content/pm/ActivityInfo;)Z
+ 
+     move-result v2
+ 
+     if-nez v2, :cond_miui_0
+ 
+     const/4 v2, 0x5
+ 
+     return v2
+ 
+     :cond_miui_0
      if-eqz v12, :cond_1
  
      move-object/from16 v0, p0
***************
*** 14970,14972 ****
--- 14998,15040 ----
  
      return-void
  .end method
+ 
+ .method private checkRunningCompatibility(Landroid/content/pm/ActivityInfo;)Z
+     .locals 2
+     .parameter "aInfo"
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/am/ActivityStack;->mService:Lcom/android/server/am/ActivityManagerService;
+ 
+     iget-boolean v0, v0, Lcom/android/server/am/ActivityManagerService;->mSystemReady:Z
+ 
+     if-eqz v0, :cond_1
+ 
+     iget-object v1, p0, Lcom/android/server/am/ActivityStack;->mContext:Landroid/content/Context;
+ 
+     if-eqz p1, :cond_0
+ 
+     iget-object v0, p1, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     :goto_0
+     invoke-static {v1, v0}, Lcom/android/server/am/ExtraActivityManagerService;->checkRunningCompatibility(Landroid/content/Context;Landroid/content/pm/ApplicationInfo;)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_1
+ 
+     const/4 v0, 0x0
+ 
+     :goto_1
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ 
+     :cond_1
+     const/4 v0, 0x1
+ 
+     goto :goto_1
+ .end method
