*** ./smali/com/android/server/am/AppNotRespondingDialog$1.smali	2015-04-19 13:47:37.109103442 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/am/AppNotRespondingDialog$1.smali	2015-04-19 13:49:06.553106782 +0800
***************
*** 36,43 ****
--- 36,62 ----
  .method public handleMessage(Landroid/os/Message;)V
      .locals 8
      .parameter "msg"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
+     const/4 v5, 0x0
+ 
+     iget-object v3, p0, Lcom/android/server/am/AppNotRespondingDialog$1;->this$0:Lcom/android/server/am/AppNotRespondingDialog;
+ 
+     invoke-virtual {v3}, Lcom/android/server/am/AppNotRespondingDialog;->getContext()Landroid/content/Context;
+ 
+     move-result-object v3
+ 
+     iget-object v4, p0, Lcom/android/server/am/AppNotRespondingDialog$1;->this$0:Lcom/android/server/am/AppNotRespondingDialog;
+ 
+     invoke-virtual {v4}, Lcom/android/server/am/AppNotRespondingDialog;->getProc()Lcom/android/server/am/ProcessRecord;
+ 
+     move-result-object v4
+ 
+     invoke-static {v3, v4, v5}, Lcom/android/server/am/MiuiErrorReport;->sendAnrErrorReport(Landroid/content/Context;Lcom/android/server/am/ProcessRecord;Z)V
+ 
      const/4 v1, 0x0
  
      .local v1, appErrorIntent:Landroid/content/Intent;
