*** ./smali/com/android/server/am/AppNotRespondingDialog.smali	2015-04-19 13:47:37.569103459 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/am/AppNotRespondingDialog.smali	2015-04-19 13:49:06.921106796 +0800
***************
*** 300,305 ****
--- 300,317 ----
  
  
  # virtual methods
+ .method getProc()Lcom/android/server/am/ProcessRecord;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/am/AppNotRespondingDialog;->mProc:Lcom/android/server/am/ProcessRecord;
+ 
+     return-object v0
+ .end method
+ 
  .method public onStop()V
      .locals 0
  
