*** ./smali/android/app/ActivityThread.smali	2015-04-19 13:46:28.169100867 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/ActivityThread.smali	2015-04-19 13:47:54.125104077 +0800
***************
*** 9271,9276 ****
--- 9271,9279 ----
      .locals 8
      .parameter "config"
      .parameter "compat"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v5, 0x1
***************
*** 9336,9341 ****
--- 9339,9346 ----
      or-int/lit16 v0, v0, 0xd00
  
      :cond_3
+     invoke-static {v0, p1}, Landroid/app/MiuiThemeHelper;->handleExtraConfigurationChanges(ILandroid/content/res/Configuration;)V
+ 
      iget-object v7, p1, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
  
      if-eqz v7, :cond_4
***************
*** 10614,10619 ****
--- 10619,10627 ----
      .locals 9
      .parameter "resDir"
      .parameter "compInfo"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v1, 0x0
***************
*** 10701,10714 ****
      move-result-object v3
  
      .local v3, metrics:Landroid/util/DisplayMetrics;
!     new-instance v4, Landroid/content/res/Resources;
  
      .end local v4           #r:Landroid/content/res/Resources;
      invoke-virtual {p0}, Landroid/app/ActivityThread;->getConfiguration()Landroid/content/res/Configuration;
  
      move-result-object v6
  
!     invoke-direct {v4, v0, v3, v6, p2}, Landroid/content/res/Resources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;Landroid/content/res/CompatibilityInfo;)V
  
      .restart local v4       #r:Landroid/content/res/Resources;
      iget-object v7, p0, Landroid/app/ActivityThread;->mPackages:Ljava/util/HashMap;
--- 10709,10722 ----
      move-result-object v3
  
      .local v3, metrics:Landroid/util/DisplayMetrics;
!     new-instance v4, Landroid/content/res/MiuiResources;
  
      .end local v4           #r:Landroid/content/res/Resources;
      invoke-virtual {p0}, Landroid/app/ActivityThread;->getConfiguration()Landroid/content/res/Configuration;
  
      move-result-object v6
  
!     invoke-direct {v4, v0, v3, v6, p2}, Landroid/content/res/MiuiResources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;Landroid/content/res/CompatibilityInfo;)V
  
      .restart local v4       #r:Landroid/content/res/Resources;
      iget-object v7, p0, Landroid/app/ActivityThread;->mPackages:Ljava/util/HashMap;
***************
*** 10808,10813 ****
--- 10816,10871 ----
      goto :goto_1
  .end method
  
+ .method getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/app/LoadedApk;)Landroid/content/res/Resources;
+     .locals 1
+     .parameter "packageName"
+     .parameter "resDir"
+     .parameter "pkgInfo"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p3, Landroid/app/LoadedApk;->mCompatibilityInfo:Landroid/view/CompatibilityInfoHolder;
+ 
+     invoke-virtual {v0}, Landroid/view/CompatibilityInfoHolder;->get()Landroid/content/res/CompatibilityInfo;
+ 
+     move-result-object v0
+ 
+     invoke-virtual {p0, p1, p2, v0}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
+     .locals 2
+     .parameter "packageName"
+     .parameter "resDir"
+     .parameter "compInfo"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0, p2, p3}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
+ 
+     move-result-object v0
+ 
+     .local v0, r:Landroid/content/res/Resources;
+     if-eqz v0, :cond_miui_0
+ 
+     move-object v1, v0
+ 
+     check-cast v1, Landroid/content/res/MiuiResources;
+ 
+     invoke-virtual {v1, p1}, Landroid/content/res/MiuiResources;->init(Ljava/lang/String;)V
+ 
+     :cond_miui_0
+     return-object v0
+ .end method
+ 
  .method final handleActivityConfigurationChanged(Landroid/os/IBinder;)V
      .locals 3
      .parameter "token"
