*** ./smali/android/app/Activity.smali	2015-04-19 13:46:24.145100717 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/Activity.smali	2015-04-19 13:47:52.997104035 +0800
***************
*** 16,22 ****
          Landroid/app/Activity$1;,
          Landroid/app/Activity$ManagedCursor;,
          Landroid/app/Activity$NonConfigurationInstances;,
!         Landroid/app/Activity$ManagedDialog;
      }
  .end annotation
  
--- 16,23 ----
          Landroid/app/Activity$1;,
          Landroid/app/Activity$ManagedCursor;,
          Landroid/app/Activity$NonConfigurationInstances;,
!         Landroid/app/Activity$ManagedDialog;,
!         Landroid/app/Activity$Injector;
      }
  .end annotation
  
***************
*** 353,361 ****
      return-void
  
      :cond_1
!     new-instance v1, Lcom/android/internal/app/ActionBarImpl;
  
!     invoke-direct {v1, p0}, Lcom/android/internal/app/ActionBarImpl;-><init>(Landroid/app/Activity;)V
  
      iput-object v1, p0, Landroid/app/Activity;->mActionBar:Lcom/android/internal/app/ActionBarImpl;
  
--- 354,362 ----
      return-void
  
      :cond_1
!     invoke-static {p0}, Landroid/app/Activity$Injector;->generateActionBar(Landroid/app/Activity;)Lcom/android/internal/app/ActionBarImpl;
  
!     move-result-object v1
  
      iput-object v1, p0, Landroid/app/Activity;->mActionBar:Lcom/android/internal/app/ActionBarImpl;
  
***************
*** 2287,2292 ****
--- 2288,2322 ----
      goto :goto_0
  .end method
  
+ .method public getMiuiActionBar()Lmiui/v5/app/MiuiActionBar;
+     .locals 2
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0}, Landroid/app/Activity;->getActionBar()Landroid/app/ActionBar;
+ 
+     move-result-object v0
+ 
+     .local v0, bar:Landroid/app/ActionBar;
+     instance-of v1, v0, Lmiui/v5/app/MiuiActionBar;
+ 
+     if-eqz v1, :cond_0
+ 
+     check-cast v0, Lmiui/v5/app/MiuiActionBar;
+ 
+     .end local v0           #bar:Landroid/app/ActionBar;
+     :goto_0
+     return-object v0
+ 
+     .restart local v0       #bar:Landroid/app/ActionBar;
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public final getParent()Landroid/app/Activity;
      .locals 1
  
***************
*** 2490,2495 ****
--- 2520,2537 ----
      return v0
  .end method
  
+ .method getToken()Landroid/os/IBinder;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/app/Activity;->mToken:Landroid/os/IBinder;
+ 
+     return-object v0
+ .end method
+ 
  .method public final getVolumeControlStream()I
      .locals 1
  
***************
*** 3250,3255 ****
--- 3292,3299 ----
  
      iput-boolean v3, p0, Landroid/app/Activity;->mCalled:Z
  
+     invoke-static {p0}, Landroid/app/Activity$Injector;->setActivityGravity(Landroid/app/Activity;)V
+ 
      return-void
  
      :cond_3
***************
*** 4553,4559 ****
      return v0
  
      :cond_0
!     const/4 v0, 0x0
  
      goto :goto_0
  .end method
--- 4597,4605 ----
      return v0
  
      :cond_0
!     invoke-static {p0, p1}, Landroid/app/Activity$Injector;->onOptionsItemSelected(Landroid/app/Activity;Landroid/view/MenuItem;)Z
! 
!     move-result v0
  
      goto :goto_0
  .end method
***************
*** 4832,4837 ****
--- 4878,4886 ----
  
  .method protected onResume()V
      .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-virtual {p0}, Landroid/app/Activity;->getApplication()Landroid/app/Application;
***************
*** 4844,4849 ****
--- 4893,4900 ----
  
      iput-boolean v0, p0, Landroid/app/Activity;->mCalled:Z
  
+     invoke-static {p0}, Landroid/app/Activity$Injector;->checkAccessControl(Landroid/app/Activity;)V
+ 
      return-void
  .end method
  
***************
*** 5292,5297 ****
--- 5343,5350 ----
      .prologue
      const/4 v2, 0x0
  
+     invoke-static {p0}, Landroid/app/Activity$Injector;->onWindowHide(Landroid/app/Activity;)V
+ 
      iget-object v0, p0, Landroid/app/Activity;->mFragments:Landroid/app/FragmentManagerImpl;
  
      invoke-virtual {v0}, Landroid/app/FragmentManagerImpl;->dispatchPause()V
***************
*** 5704,5709 ****
--- 5757,5764 ----
      throw v0
  
      :cond_1
+     invoke-static {p0}, Landroid/app/Activity$Injector;->onWindowShow(Landroid/app/Activity;)V
+ 
      return-void
  .end method
  
