*** ./smali/android/app/ActivityManager.smali	2015-04-19 13:46:22.977100673 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/ActivityManager.smali	2015-04-19 13:47:52.029103999 +0800
***************
*** 69,74 ****
--- 69,76 ----
  
  .field public static final START_FORWARD_AND_REQUEST_CONFLICT:I = -0x3
  
+ .field public static final START_INCOMPATIBLE:I = 0x5
+ 
  .field public static final START_INTENT_NOT_RESOLVED:I = -0x1
  
  .field public static final START_NOT_ACTIVITY:I = -0x5
***************
*** 256,261 ****
--- 258,266 ----
  .method public static isHighEndGfx(Landroid/view/Display;)Z
      .locals 8
      .parameter "display"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v3, 0x1
***************
*** 300,306 ****
  
      if-ge v1, v4, :cond_0
  
!     const/4 v3, 0x0
  
      goto :goto_0
  .end method
--- 305,313 ----
  
      if-ge v1, v4, :cond_0
  
!     invoke-static {p0}, Landroid/app/ExtraActivityManager;->isHighEndGfx(Landroid/view/Display;)Z
! 
!     move-result v3
  
      goto :goto_0
  .end method
