*** ./smali/android/app/ApplicationPackageManager.smali	2015-04-19 13:46:22.765100665 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/ApplicationPackageManager.smali	2015-04-19 13:47:51.825103992 +0800
***************
*** 1394,1410 ****
      goto :goto_0
  .end method
  
  .method public getDefaultActivityIcon()Landroid/graphics/drawable/Drawable;
      .locals 2
  
      .prologue
!     invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
! 
!     move-result-object v0
  
!     const v1, 0x1080093
  
!     invoke-virtual {v0, v1}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v0
  
--- 1394,1423 ----
      goto :goto_0
  .end method
  
+ .method getContext()Landroid/app/ContextImpl;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/app/ApplicationPackageManager;->mContext:Landroid/app/ContextImpl;
+ 
+     return-object v0
+ .end method
+ 
  .method public getDefaultActivityIcon()Landroid/graphics/drawable/Drawable;
      .locals 2
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
!     iget-object v0, p0, Landroid/app/ApplicationPackageManager;->mContext:Landroid/app/ContextImpl;
  
!     const-string v1, "sym_def_app_icon.png"
  
!     invoke-static {v0, v1}, Lmiui/content/res/IconCustomizer;->getCustomizedIcon(Landroid/content/Context;Ljava/lang/String;)Landroid/graphics/drawable/BitmapDrawable;
  
      move-result-object v0
  
***************
*** 2439,2444 ****
--- 2452,2461 ----
  .method public getResourcesForApplication(Landroid/content/pm/ApplicationInfo;)Landroid/content/res/Resources;
      .locals 4
      .parameter "app"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Landroid/content/pm/PackageManager$NameNotFoundException;
***************
*** 2476,2481 ****
--- 2493,2500 ----
  
      iget-object v2, v1, Landroid/app/ContextImpl;->mMainThread:Landroid/app/ActivityThread;
  
+     iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
+ 
      iget v1, p1, Landroid/content/pm/ApplicationInfo;->uid:I
  
      invoke-static {}, Landroid/os/Process;->myUid()I
***************
*** 2491,2497 ****
  
      iget-object v3, v3, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
  
!     invoke-virtual {v2, v1, v3}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Landroid/app/LoadedApk;)Landroid/content/res/Resources;
  
      move-result-object v0
  
--- 2510,2516 ----
  
      iget-object v3, v3, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
  
!     invoke-virtual {v2, v0, v1, v3}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/app/LoadedApk;)Landroid/content/res/Resources;
  
      move-result-object v0
  
