*** ./smali/android/app/ContextImpl.smali	2015-04-19 13:46:23.193100681 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/ContextImpl.smali	2015-04-19 13:47:52.253104007 +0800
***************
*** 8,14 ****
      value = {
          Landroid/app/ContextImpl$ApplicationContentResolver;,
          Landroid/app/ContextImpl$StaticServiceFetcher;,
!         Landroid/app/ContextImpl$ServiceFetcher;
      }
  .end annotation
  
--- 8,15 ----
      value = {
          Landroid/app/ContextImpl$ApplicationContentResolver;,
          Landroid/app/ContextImpl$StaticServiceFetcher;,
!         Landroid/app/ContextImpl$ServiceFetcher;,
!         Landroid/app/ContextImpl$Injector;
      }
  .end annotation
  
***************
*** 427,432 ****
--- 428,435 ----
  
      invoke-static {v0, v1}, Landroid/app/ContextImpl;->registerService(Ljava/lang/String;Landroid/app/ContextImpl$ServiceFetcher;)V
  
+     invoke-static {}, Landroid/app/ContextImpl;->registerMiuiServices()V
+ 
      return-void
  .end method
  
***************
*** 1082,1087 ****
--- 1085,1105 ----
      throw v0
  .end method
  
+ .method static registerMiuiServices()V
+     .locals 2
+ 
+     .prologue
+     const-string v0, "security"
+ 
+     new-instance v1, Landroid/app/ContextImpl$SecurityServiceFetcher;
+ 
+     invoke-direct {v1}, Landroid/app/ContextImpl$SecurityServiceFetcher;-><init>()V
+ 
+     invoke-static {v0, v1}, Landroid/app/ContextImpl;->registerService(Ljava/lang/String;Landroid/app/ContextImpl$ServiceFetcher;)V
+ 
+     return-void
+ .end method
+ 
  .method private registerReceiverInternal(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;Ljava/lang/String;Landroid/os/Handler;Landroid/content/Context;)Landroid/content/Intent;
      .locals 8
      .parameter "receiver"
***************
*** 1089,1094 ****
--- 1107,1115 ----
      .parameter "broadcastPermission"
      .parameter "scheduler"
      .parameter "context"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v7, 0x0
***************
*** 1137,1142 ****
--- 1158,1165 ----
      .restart local v3       #rd:Landroid/content/IIntentReceiver;
      :cond_1
      :goto_0
+     invoke-static {p0, p2}, Landroid/app/ContextImpl$Injector;->checkPriority(Landroid/app/ContextImpl;Landroid/content/IntentFilter;)V
+ 
      :try_start_0
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 3475,3486 ****
  .end method
  
  .method final init(Landroid/app/LoadedApk;Landroid/os/IBinder;Landroid/app/ActivityThread;Landroid/content/res/Resources;Ljava/lang/String;)V
!     .locals 2
      .parameter "packageInfo"
      .parameter "activityToken"
      .parameter "mainThread"
      .parameter "container"
      .parameter "basePackageName"
  
      .prologue
      iput-object p1, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
--- 3498,3512 ----
  .end method
  
  .method final init(Landroid/app/LoadedApk;Landroid/os/IBinder;Landroid/app/ActivityThread;Landroid/content/res/Resources;Ljava/lang/String;)V
!     .locals 3
      .parameter "packageInfo"
      .parameter "activityToken"
      .parameter "mainThread"
      .parameter "container"
      .parameter "basePackageName"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iput-object p1, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
***************
*** 3533,3539 ****
  
      move-result-object v1
  
!     invoke-virtual {p3, v0, v1}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
  
      move-result-object v0
  
--- 3559,3569 ----
  
      move-result-object v1
  
!     iget-object v2, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
! 
!     iget-object v2, v2, Landroid/app/LoadedApk;->mPackageName:Ljava/lang/String;
! 
!     invoke-virtual {p3, v2, v0, v1}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
  
      move-result-object v0
  
