*** ./smali/android/app/DownloadManager$Query.smali	2015-04-19 13:46:29.685100924 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/DownloadManager$Query.smali	2015-04-19 13:47:55.629104134 +0800
***************
*** 170,175 ****
--- 170,197 ----
  
  
  # virtual methods
+ .method addExtraSelectionParts(Ljava/util/List;)V
+     .locals 0
+     .parameter
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "(",
+             "Ljava/util/List",
+             "<",
+             "Ljava/lang/String;",
+             ">;)V"
+         }
+     .end annotation
+ 
+     .prologue
+     .local p1, selectionParts:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
+     return-void
+ .end method
+ 
  .method public orderBy(Ljava/lang/String;I)Landroid/app/DownloadManager$Query;
      .locals 3
      .parameter "column"
***************
*** 272,277 ****
--- 294,302 ----
      .parameter "resolver"
      .parameter "projection"
      .parameter "baseUri"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      move-object v1, p3
***************
*** 496,501 ****
--- 521,530 ----
  
      move-result-object v0
  
+     invoke-static {v0}, Landroid/app/DownloadManager$Injector;->addParens(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
      invoke-interface {v8, v0}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
      .end local v7           #parts:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
***************
*** 509,514 ****
--- 538,545 ----
      invoke-interface {v8, v0}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
      :cond_7
+     invoke-virtual {p0, v8}, Landroid/app/DownloadManager$Query;->addExtraSelectionParts(Ljava/util/List;)V
+ 
      const-string v0, "deleted != \'1\'"
  
      invoke-interface {v8, v0}, Ljava/util/List;->add(Ljava/lang/Object;)Z
***************
*** 606,608 ****
--- 637,665 ----
  
      return-object p0
  .end method
+ 
+ .method setOrderByColumn(Ljava/lang/String;)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-object p1, p0, Landroid/app/DownloadManager$Query;->mOrderByColumn:Ljava/lang/String;
+ 
+     return-void
+ .end method
+ 
+ .method setOrderDirection(I)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput p1, p0, Landroid/app/DownloadManager$Query;->mOrderDirection:I
+ 
+     return-void
+ .end method
