*** ./smali/android/app/LoadedApk.smali	2015-04-19 13:46:25.541100769 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/LoadedApk.smali	2015-04-19 13:47:53.673104061 +0800
***************
*** 1753,1769 ****
  .end method
  
  .method public getResources(Landroid/app/ActivityThread;)Landroid/content/res/Resources;
!     .locals 1
      .parameter "mainThread"
  
      .prologue
      iget-object v0, p0, Landroid/app/LoadedApk;->mResources:Landroid/content/res/Resources;
  
      if-nez v0, :cond_0
  
!     iget-object v0, p0, Landroid/app/LoadedApk;->mResDir:Ljava/lang/String;
  
!     invoke-virtual {p1, v0, p0}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Landroid/app/LoadedApk;)Landroid/content/res/Resources;
  
      move-result-object v0
  
--- 1753,1774 ----
  .end method
  
  .method public getResources(Landroid/app/ActivityThread;)Landroid/content/res/Resources;
!     .locals 2
      .parameter "mainThread"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iget-object v0, p0, Landroid/app/LoadedApk;->mResources:Landroid/content/res/Resources;
  
      if-nez v0, :cond_0
  
!     iget-object v0, p0, Landroid/app/LoadedApk;->mPackageName:Ljava/lang/String;
  
!     iget-object v1, p0, Landroid/app/LoadedApk;->mResDir:Ljava/lang/String;
! 
!     invoke-virtual {p1, v0, v1, p0}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/app/LoadedApk;)Landroid/content/res/Resources;
  
      move-result-object v0
  
