*** ./smali/android/app/Notification.smali	2015-04-19 13:46:23.553100695 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/Notification.smali	2015-04-19 13:47:52.389104013 +0800
***************
*** 95,100 ****
--- 95,102 ----
  
  .field public deleteIntent:Landroid/app/PendingIntent;
  
+ .field public extraNotification:Lmiui/app/ExtraNotification;
+ 
  .field private extras:Landroid/os/Bundle;
  
  .field public flags:I
***************
*** 150,155 ****
--- 152,163 ----
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v0}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 178,183 ****
--- 186,197 ----
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v0}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 206,211 ****
--- 220,231 ----
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v0}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 232,237 ****
--- 252,263 ----
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v1, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v1}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v1, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v1, -0x1
  
      iput v1, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 495,500 ****
--- 521,530 ----
      iput-object v1, p0, Landroid/app/Notification;->bigContentView:Landroid/widget/RemoteViews;
  
      :cond_9
+     iget-object v1, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
+     invoke-virtual {v1, p1}, Lmiui/app/ExtraNotification;->readFromParcel(Landroid/os/Parcel;)V
+ 
      return-void
  .end method
  
***************
*** 752,757 ****
--- 782,793 ----
      iput-object v7, v2, Landroid/app/Notification;->bigContentView:Landroid/widget/RemoteViews;
  
      :cond_8
+     iget-object v7, v2, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
+     iget-object v8, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
+     invoke-virtual {v7, v8}, Lmiui/app/ExtraNotification;->setTo(Lmiui/app/ExtraNotification;)V
+ 
      return-object v2
  .end method
  
***************
*** 1363,1368 ****
--- 1399,1408 ----
      invoke-virtual {v0, p1, v2}, Landroid/widget/RemoteViews;->writeToParcel(Landroid/os/Parcel;I)V
  
      :goto_9
+     iget-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
+     invoke-virtual {v0, p1, p2}, Lmiui/app/ExtraNotification;->writeToParcel(Landroid/os/Parcel;I)V
+ 
      return-void
  
      :cond_0
