*** ./smali/android/app/ProgressDialog.smali	2015-04-19 13:46:23.725100701 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/app/ProgressDialog.smali	2015-04-19 13:47:52.569104019 +0800
***************
*** 641,646 ****
--- 641,652 ----
  
      invoke-virtual {p0, v2}, Landroid/app/ProgressDialog;->setView(Landroid/view/View;)V
  
+     iget-object v3, p0, Landroid/app/ProgressDialog;->mContext:Landroid/content/Context;
+ 
+     iget-object v4, p0, Landroid/app/ProgressDialog;->mMessageView:Landroid/widget/TextView;
+ 
+     invoke-static {p0, v3, v4}, Landroid/app/ProgressDialog$Injector;->usingV5Style(Landroid/app/ProgressDialog;Landroid/content/Context;Landroid/widget/TextView;)V
+ 
      goto :goto_0
  .end method
  
