*** ./smali/android/widget/AbsListView$FlingRunnable.smali	2015-04-19 13:46:20.137100567 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/widget/AbsListView$FlingRunnable.smali	2015-04-19 13:47:49.377103900 +0800
***************
*** 70,75 ****
--- 70,131 ----
      return-object v0
  .end method
  
+ .method private endFling2()V
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
+     invoke-virtual {v0}, Landroid/widget/AbsListView;->getScrollY()I
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {p0}, Landroid/widget/AbsListView$FlingRunnable;->startSpringback()V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_0
+     invoke-virtual {p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling()V
+ 
+     goto :goto_0
+ .end method
+ 
+ .method private scaleOverScrollDeltaY(II)I
+     .locals 2
+     .parameter "deltaY"
+     .parameter "scrollY"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     move v0, p1
+ 
+     .local v0, retval:I
+     iget-object v1, p0, Landroid/widget/AbsListView$FlingRunnable;->mScroller:Landroid/widget/OverScroller;
+ 
+     invoke-virtual {v1}, Landroid/widget/OverScroller;->checkSpringBackState()Z
+ 
+     move-result v1
+ 
+     if-nez v1, :cond_0
+ 
+     iget-object v1, p0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
+     invoke-static {v1, p1, p2}, Landroid/widget/AbsListView$Injector;->scaleOverScrollDeltaY(Landroid/widget/AbsListView;II)I
+ 
+     move-result v0
+ 
+     :cond_0
+     return v0
+ .end method
+ 
  
  # virtual methods
  .method edgeReached(I)V
***************
*** 271,276 ****
--- 327,344 ----
      return-void
  .end method
  
+ .method protected getScroller()Landroid/widget/OverScroller;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/AbsListView$FlingRunnable;->mScroller:Landroid/widget/OverScroller;
+ 
+     return-object v0
+ .end method
+ 
  .method public run()V
      .locals 30
  
***************
*** 506,512 ****
  
      sub-int v2, v20, v2
  
!     neg-int v4, v2
  
      .local v4, overshoot:I
      move-object/from16 v0, p0
--- 575,597 ----
  
      sub-int v2, v20, v2
  
!     neg-int v2, v2
! 
!     move-object/from16 v0, p0
! 
!     iget-object v3, v0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
! 
!     move-object/from16 v0, p0
! 
!     iget-object v5, v0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
! 
!     invoke-virtual {v5}, Landroid/widget/AbsListView;->getScrollY()I
! 
!     move-result v5
! 
!     invoke-static {v3, v2, v5}, Landroid/widget/AbsListView$Injector;->scaleOverScrollDeltaY(Landroid/widget/AbsListView;II)I
! 
!     move-result v4
  
      .local v4, overshoot:I
      move-object/from16 v0, p0
***************
*** 731,737 ****
      move-result v19
  
      .local v19, currY:I
!     sub-int v7, v19, v9
  
      .local v7, deltaY:I
      move-object/from16 v0, p0
--- 816,828 ----
      move-result v19
  
      .local v19, currY:I
!     sub-int v2, v19, v9
! 
!     move-object/from16 v0, p0
! 
!     invoke-direct {v0, v2, v9}, Landroid/widget/AbsListView$FlingRunnable;->scaleOverScrollDeltaY(II)I
! 
!     move-result v7
  
      .local v7, deltaY:I
      move-object/from16 v0, p0
***************
*** 855,861 ****
      .end local v9           #scrollY:I
      .end local v19           #currY:I
      :cond_11
!     invoke-virtual/range {p0 .. p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling()V
  
      goto/16 :goto_0
  
--- 946,952 ----
      .end local v9           #scrollY:I
      .end local v19           #currY:I
      :cond_11
!     invoke-direct/range {p0 .. p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling2()V
  
      goto/16 :goto_0
  
