*** ./smali/android/widget/TimePicker.smali	2015-04-19 13:46:21.145100605 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/widget/TimePicker.smali	2015-04-19 13:47:50.389103938 +0800
***************
*** 7,13 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/widget/TimePicker$SavedState;,
!         Landroid/widget/TimePicker$OnTimeChangedListener;
      }
  .end annotation
  
--- 7,14 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/widget/TimePicker$SavedState;,
!         Landroid/widget/TimePicker$OnTimeChangedListener;,
!         Landroid/widget/TimePicker$OnMinuteChangeListener;
      }
  .end annotation
  
***************
*** 96,101 ****
--- 97,105 ----
      .parameter "context"
      .parameter "attrs"
      .parameter "defStyle"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v11, 0x5
***************
*** 240,245 ****
--- 244,257 ----
  
      iget-object v4, p0, Landroid/widget/TimePicker;->mMinuteSpinner:Landroid/widget/NumberPicker;
  
+     new-instance v5, Landroid/widget/TimePicker$OnMinuteChangeListener;
+ 
+     invoke-direct {v5, p0}, Landroid/widget/TimePicker$OnMinuteChangeListener;-><init>(Landroid/widget/TimePicker;)V
+ 
+     invoke-virtual {v4, v5}, Landroid/widget/NumberPicker;->setOnValueChangedListener(Landroid/widget/NumberPicker$OnValueChangeListener;)V
+ 
+     iget-object v4, p0, Landroid/widget/TimePicker;->mMinuteSpinner:Landroid/widget/NumberPicker;
+ 
      invoke-virtual {v4, v9}, Landroid/widget/NumberPicker;->findViewById(I)Landroid/view/View;
  
      move-result-object v4
***************
*** 470,475 ****
--- 482,521 ----
      return-object v0
  .end method
  
+ .method private changePositionState(Z)V
+     .locals 2
+     .parameter "is24HourView"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/TimePicker;->mMinuteSpinner:Landroid/widget/NumberPicker;
+ 
+     instance-of v0, v0, Lmiui/widget/NumberPicker;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Landroid/widget/TimePicker;->mMinuteSpinner:Landroid/widget/NumberPicker;
+ 
+     check-cast v0, Lmiui/widget/NumberPicker;
+ 
+     if-eqz p1, :cond_1
+ 
+     const/4 v1, 0x1
+ 
+     :goto_0
+     invoke-virtual {v0, v1}, Lmiui/widget/NumberPicker;->setPositionState(I)V
+ 
+     :cond_0
+     return-void
+ 
+     :cond_1
+     const/4 v1, 0x3
+ 
+     goto :goto_0
+ .end method
+ 
  .method private onTimeChanged()V
      .locals 3
  
***************
*** 827,832 ****
--- 873,902 ----
  
  
  # virtual methods
+ .method callOnTimeChanged()V
+     .locals 0
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_CLASS:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0}, Landroid/widget/TimePicker;->onTimeChanged()V
+ 
+     return-void
+ .end method
+ 
+ .method callUpdateInputState()V
+     .locals 0
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0}, Landroid/widget/TimePicker;->updateInputState()V
+ 
+     return-void
+ .end method
+ 
  .method public dispatchPopulateAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)Z
      .locals 1
      .parameter "event"
***************
*** 1323,1328 ****
--- 1393,1404 ----
  
      iput-boolean v1, p0, Landroid/widget/TimePicker;->mIs24HourView:Z
  
+     invoke-virtual {p1}, Ljava/lang/Boolean;->booleanValue()Z
+ 
+     move-result v1
+ 
+     invoke-direct {p0, v1}, Landroid/widget/TimePicker;->changePositionState(Z)V
+ 
      invoke-virtual {p0}, Landroid/widget/TimePicker;->getCurrentHour()Ljava/lang/Integer;
  
      move-result-object v1
