*** ./smali/com/android/server/wm/WindowManagerService.smali	2015-04-19 13:47:31.661103238 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowManagerService.smali	2015-04-19 13:49:01.381106589 +0800
***************
*** 5403,5408 ****
--- 5403,5417 ----
  
      .end local v3           #token:Lcom/android/server/wm/WindowToken;
      :cond_3
+     invoke-static {p0, p2}, Lcom/android/server/wm/WindowManagerService$Injector;->updateFocusAndLayout(Lcom/android/server/wm/WindowManagerService;I)Z
+ 
+     move-result v4
+ 
+     if-nez v4, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      iget-object v4, p0, Lcom/android/server/wm/WindowManagerService;->mInputMonitor:Lcom/android/server/wm/InputMonitor;
  
      invoke-virtual {v4}, Lcom/android/server/wm/InputMonitor;->setUpdateInputWindowsNeededLw()V
***************
*** 5913,5918 ****
--- 5922,5929 ----
      .end local v0           #N:I
      :cond_6
      :goto_7
+     invoke-static {p0}, Lcom/android/server/wm/WindowManagerService$Injector;->hideFloatingWindow(Lcom/android/server/wm/WindowManagerService;)V
+ 
      invoke-static {v11, v12}, Landroid/os/Trace;->traceEnd(J)V
  
      goto/16 :goto_0
***************
*** 15411,15416 ****
--- 15422,15431 ----
      :cond_2
      iput-boolean v5, p0, Lcom/android/server/wm/WindowManagerService;->mSafeMode:Z
  
+     const/4 v6, 0x0
+ 
+     iput-boolean v6, p0, Lcom/android/server/wm/WindowManagerService;->mSafeMode:Z
+ 
      :try_start_0
      const-string v5, "persist.sys.safemode"
  
***************
*** 25462,25467 ****
--- 25477,25484 ----
      return v29
  
      :cond_1
+     invoke-static/range {v27 .. v27}, Lcom/android/server/wm/WindowManagerService$Injector;->restoreFloatingWindowState(Lcom/android/server/wm/WindowState;)V
+ 
      move-object/from16 v0, v27
  
      iget-object v0, v0, Lcom/android/server/wm/WindowState;->mWinAnimator:Lcom/android/server/wm/WindowStateAnimator;
***************
*** 26597,26602 ****
--- 26614,26621 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/wm/InputMonitor;->updateInputWindowsLw(Z)V
  
+     invoke-static/range {v27 .. v27}, Lcom/android/server/wm/WindowManagerService$Injector;->saveFloatingWindowState(Lcom/android/server/wm/WindowState;)V
+ 
      monitor-exit v30
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
***************
*** 26805,26810 ****
--- 26824,26831 ----
  
      invoke-static/range {v17 .. v18}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     invoke-static/range {v27 .. v27}, Lcom/android/server/wm/WindowManagerService$Injector;->saveFloatingWindowState(Lcom/android/server/wm/WindowState;)V
+ 
      const/16 v29, 0x0
  
      monitor-exit v30
