*** ./smali/com/android/server/wm/WindowStateAnimator.smali	2015-04-19 13:47:31.577103235 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowStateAnimator.smali	2015-04-19 13:49:01.305106586 +0800
***************
*** 104,109 ****
--- 104,115 ----
  
  .field mLocalAnimating:Z
  
+ .field mMiuiHidden:Z
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field mPendingDestroySurface:Landroid/view/Surface;
  
  .field final mPolicy:Landroid/view/WindowManagerPolicy;
***************
*** 4280,4288 ****
  
      iput-boolean v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceShown:Z
  
!     iget-object v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurface:Landroid/view/Surface;
! 
!     invoke-virtual {v3}, Landroid/view/Surface;->show()V
  
      iget-object v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mWin:Lcom/android/server/wm/WindowState;
  
--- 4286,4292 ----
  
      iput-boolean v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceShown:Z
  
!     invoke-static {p0}, Lcom/android/server/wm/WindowStateAnimator$Injector;->showSurface(Lcom/android/server/wm/WindowStateAnimator;)V
  
      iget-object v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mWin:Lcom/android/server/wm/WindowState;
  
