*** ./smali/com/android/server/pm/PackageManagerService$PackageHandler.smali	2015-04-19 13:47:30.333103189 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/pm/PackageManagerService$PackageHandler.smali	2015-04-19 13:49:00.069106540 +0800
***************
*** 1716,1727 ****
  .method public handleMessage(Landroid/os/Message;)V
      .locals 2
      .parameter "msg"
  
      .prologue
      const/16 v1, 0xa
  
      :try_start_0
!     invoke-virtual {p0, p1}, Lcom/android/server/pm/PackageManagerService$PackageHandler;->doHandleMessage(Landroid/os/Message;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
--- 1716,1732 ----
  .method public handleMessage(Landroid/os/Message;)V
      .locals 2
      .parameter "msg"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/16 v1, 0xa
  
      :try_start_0
!     iget-object v0, p0, Lcom/android/server/pm/PackageManagerService$PackageHandler;->this$0:Lcom/android/server/pm/PackageManagerService;
! 
!     invoke-static {v0, p0, p1}, Lcom/android/server/pm/PackageManagerService$Injector;->doHandleMessage(Lcom/android/server/pm/PackageManagerService;Lcom/android/server/pm/PackageManagerService$PackageHandler;Landroid/os/Message;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
