*** ./smali/com/android/server/pm/ShutdownThread.smali	2015-04-19 13:47:30.041103178 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/pm/ShutdownThread.smali	2015-04-19 13:48:59.749106528 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/pm/ShutdownThread$CloseDialogReceiver;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/pm/ShutdownThread$CloseDialogReceiver;,
!         Lcom/android/server/pm/ShutdownThread$Injector;
      }
  .end annotation
  
***************
*** 109,114 ****
--- 110,118 ----
  .method private static beginShutdownSequence(Landroid/content/Context;)V
      .locals 7
      .parameter "context"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v5, 0x1
***************
*** 151,157 ****
      invoke-direct {v1, p0}, Landroid/app/ProgressDialog;-><init>(Landroid/content/Context;)V
  
      .local v1, pd:Landroid/app/ProgressDialog;
!     const v2, 0x104012a
  
      invoke-virtual {p0, v2}, Landroid/content/Context;->getText(I)Ljava/lang/CharSequence;
  
--- 155,161 ----
      invoke-direct {v1, p0}, Landroid/app/ProgressDialog;-><init>(Landroid/content/Context;)V
  
      .local v1, pd:Landroid/app/ProgressDialog;
!     const v2, 0x60c0026
  
      invoke-virtual {p0, v2}, Landroid/content/Context;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 159,165 ****
  
      invoke-virtual {v1, v2}, Landroid/app/ProgressDialog;->setTitle(Ljava/lang/CharSequence;)V
  
!     const v2, 0x104012e
  
      invoke-virtual {p0, v2}, Landroid/content/Context;->getText(I)Ljava/lang/CharSequence;
  
--- 163,169 ----
  
      invoke-virtual {v1, v2}, Landroid/app/ProgressDialog;->setTitle(Ljava/lang/CharSequence;)V
  
!     const v2, 0x60c002c
  
      invoke-virtual {p0, v2}, Landroid/content/Context;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 179,185 ****
  
      invoke-virtual {v2, v3}, Landroid/view/Window;->setType(I)V
  
!     invoke-virtual {v1}, Landroid/app/ProgressDialog;->show()V
  
      sget-object v2, Lcom/android/server/pm/ShutdownThread;->sInstance:Lcom/android/server/pm/ShutdownThread;
  
--- 183,189 ----
  
      invoke-virtual {v2, v3}, Landroid/view/Window;->setType(I)V
  
!     invoke-static {p0}, Lcom/android/server/pm/ShutdownThread$Injector;->createShutDownDialog(Landroid/content/Context;)V
  
      sget-object v2, Lcom/android/server/pm/ShutdownThread;->sInstance:Lcom/android/server/pm/ShutdownThread;
  
***************
*** 344,349 ****
--- 348,389 ----
      goto :goto_2
  .end method
  
+ .method static getIsStarted()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/server/pm/ShutdownThread;->sIsStarted:Z
+ 
+     return v0
+ .end method
+ 
+ .method static getIsStartedGuard()Ljava/lang/Object;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-object v0, Lcom/android/server/pm/ShutdownThread;->sIsStartedGuard:Ljava/lang/Object;
+ 
+     return-object v0
+ .end method
+ 
+ .method static getReboot()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/server/pm/ShutdownThread;->mReboot:Z
+ 
+     return v0
+ .end method
+ 
  .method public static reboot(Landroid/content/Context;Ljava/lang/String;Z)V
      .locals 1
      .parameter "context"
***************
*** 470,481 ****
--- 510,526 ----
      .locals 1
      .parameter "context"
      .parameter "confirm"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v0, 0x1
  
      sput-boolean v0, Lcom/android/server/pm/ShutdownThread;->mReboot:Z
  
+     const/4 v0, 0x0
+ 
      sput-boolean v0, Lcom/android/server/pm/ShutdownThread;->mRebootSafeMode:Z
  
      const/4 v0, 0x0
***************
*** 487,492 ****
--- 532,563 ----
      return-void
  .end method
  
+ .method static setReboot(Z)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sput-boolean p0, Lcom/android/server/pm/ShutdownThread;->mReboot:Z
+ 
+     return-void
+ .end method
+ 
+ .method static setRebootReason(Ljava/lang/String;)V
+     .locals 0
+     .parameter "value"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sput-object p0, Lcom/android/server/pm/ShutdownThread;->mRebootReason:Ljava/lang/String;
+ 
+     return-void
+ .end method
+ 
  .method public static shutdown(Landroid/content/Context;Z)V
      .locals 1
      .parameter "context"
***************
*** 508,513 ****
--- 579,587 ----
      .locals 7
      .parameter "context"
      .parameter "confirm"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      sget-object v5, Lcom/android/server/pm/ShutdownThread;->sIsStartedGuard:Ljava/lang/Object;
***************
*** 598,607 ****
  
      move-result-object v4
  
-     invoke-virtual {v4, v3}, Landroid/app/AlertDialog$Builder;->setMessage(I)Landroid/app/AlertDialog$Builder;
- 
-     move-result-object v4
- 
      const v5, 0x1040013
  
      new-instance v6, Lcom/android/server/pm/ShutdownThread$1;
--- 672,677 ----
***************
*** 625,630 ****
--- 695,702 ----
      move-result-object v1
  
      .local v1, dialog:Landroid/app/AlertDialog;
+     invoke-static {v1}, Lcom/android/server/pm/ShutdownThread$Injector;->setDialogTitle(Landroid/app/Dialog;)V
+ 
      iput-object v1, v0, Lcom/android/server/pm/ShutdownThread$CloseDialogReceiver;->dialog:Landroid/app/Dialog;
  
      invoke-virtual {v1, v0}, Landroid/app/AlertDialog;->setOnDismissListener(Landroid/content/DialogInterface$OnDismissListener;)V
***************
*** 639,644 ****
--- 711,718 ----
  
      invoke-virtual {v1}, Landroid/app/AlertDialog;->show()V
  
+     invoke-static {v1}, Lcom/android/server/pm/ShutdownThread$Injector;->setDialogPositiveButtonText(Landroid/app/AlertDialog;)V
+ 
      goto :goto_0
  
      .end local v0           #closer:Lcom/android/server/pm/ShutdownThread$CloseDialogReceiver;
