*** ./smali/com/android/server/pm/UserManager.smali	2015-04-19 13:47:30.485103195 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/pm/UserManager.smali	2015-04-19 13:49:00.217106546 +0800
***************
*** 1689,1695 ****
--- 1689,1705 ----
  
      monitor-enter v1
  
+     if-nez p1, :cond_0
+ 
+     const/4 v0, 0x1
+ 
      :try_start_0
+     monitor-exit v1
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
      iget-object v0, p0, Lcom/android/server/pm/UserManager;->mUserIds:[I
  
      invoke-static {v0, p1}, Lcom/android/internal/util/ArrayUtils;->contains([II)Z
***************
*** 1698,1704 ****
  
      monitor-exit v1
  
!     return v0
  
      :catchall_0
      move-exception v0
--- 1708,1714 ----
  
      monitor-exit v1
  
!     goto :goto_0
  
      :catchall_0
      move-exception v0
