*** ./smali/com/android/internal/view/menu/MenuBuilder.smali	2015-04-19 13:47:15.705102643 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/view/menu/MenuBuilder.smali	2015-04-19 13:48:43.377105917 +0800
***************
*** 2475,2480 ****
--- 2475,2528 ----
      return v0
  .end method
  
+ .method public miuiGetActionItems()Ljava/util/ArrayList;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "()",
+             "Ljava/util/ArrayList",
+             "<",
+             "Lcom/android/internal/view/menu/MenuItemImpl;",
+             ">;"
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getActionItems()Ljava/util/ArrayList;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method public miuiGetVisibleItems()Ljava/util/ArrayList;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "()",
+             "Ljava/util/ArrayList",
+             "<",
+             "Lcom/android/internal/view/menu/MenuItemImpl;",
+             ">;"
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getVisibleItems()Ljava/util/ArrayList;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
  .method onItemActionRequestChanged(Lcom/android/internal/view/menu/MenuItemImpl;)V
      .locals 1
      .parameter "item"
