*** ./smali/android/content/res/Resources.smali	2015-04-19 13:46:42.005101384 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/content/res/Resources.smali	2015-04-19 13:48:10.537104690 +0800
***************
*** 7,13 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/content/res/Resources$Theme;,
!         Landroid/content/res/Resources$NotFoundException;
      }
  .end annotation
  
--- 7,14 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/content/res/Resources$Theme;,
!         Landroid/content/res/Resources$NotFoundException;,
!         Landroid/content/res/Resources$Injector;
      }
  .end annotation
  
***************
*** 170,177 ****
      return-void
  .end method
  
! .method private constructor <init>()V
      .locals 3
  
      .prologue
      const/4 v2, 0x4
--- 171,181 ----
      return-void
  .end method
  
! .method constructor <init>()V
      .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_ACCESS:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v2, 0x4
***************
*** 522,527 ****
--- 526,553 ----
      return-void
  .end method
  
+ .method static clearPreloadedCache()V
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorStateLists:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     return-void
+ .end method
+ 
  .method private getCachedColorStateList(J)Landroid/content/res/ColorStateList;
      .locals 4
      .parameter "key"
***************
*** 661,666 ****
--- 687,695 ----
  .method private getCachedStyledAttributes(I)Landroid/content/res/TypedArray;
      .locals 5
      .parameter "len"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iget-object v3, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
***************
*** 726,732 ****
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      :cond_1
      :try_start_1
!     new-instance v0, Landroid/content/res/TypedArray;
  
      .end local v0           #attrs:Landroid/content/res/TypedArray;
      mul-int/lit8 v2, p1, 0x6
--- 755,761 ----
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      :cond_1
      :try_start_1
!     new-instance v0, Landroid/content/res/MiuiTypedArray;
  
      .end local v0           #attrs:Landroid/content/res/TypedArray;
      mul-int/lit8 v2, p1, 0x6
***************
*** 737,743 ****
  
      new-array v4, v4, [I
  
!     invoke-direct {v0, p0, v2, v4, p1}, Landroid/content/res/TypedArray;-><init>(Landroid/content/res/Resources;[I[II)V
  
      monitor-exit v3
      :try_end_1
--- 766,772 ----
  
      new-array v4, v4, [I
  
!     invoke-direct {v0, p0, v2, v4, p1}, Landroid/content/res/MiuiTypedArray;-><init>(Landroid/content/res/Resources;[I[II)V
  
      monitor-exit v3
      :try_end_1
***************
*** 787,793 ****
  .end method
  
  .method public static getSystem()Landroid/content/res/Resources;
!     .locals 3
  
      .prologue
      sget-object v2, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
--- 816,825 ----
  .end method
  
  .method public static getSystem()Landroid/content/res/Resources;
!     .locals 5
!     .annotation build Landroid/annotation/MiuiHook;
!         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
!     .end annotation
  
      .prologue
      sget-object v2, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
***************
*** 800,813 ****
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     new-instance v0, Landroid/content/res/Resources;
  
      .end local v0           #ret:Landroid/content/res/Resources;
!     invoke-direct {v0}, Landroid/content/res/Resources;-><init>()V
  
      .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      :cond_0
      monitor-exit v2
  
--- 832,853 ----
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     new-instance v0, Landroid/content/res/MiuiResources;
  
      .end local v0           #ret:Landroid/content/res/Resources;
!     invoke-direct {v0}, Landroid/content/res/MiuiResources;-><init>()V
  
      .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
+     move-object v1, v0
+ 
+     check-cast v1, Landroid/content/res/MiuiResources;
+ 
+     const/4 v4, 0x0
+ 
+     invoke-virtual {v1, v4}, Landroid/content/res/MiuiResources;->init(Ljava/lang/String;)V
+ 
      :cond_0
      monitor-exit v2
  
***************
*** 2151,2156 ****
--- 2191,2225 ----
      goto :goto_0
  .end method
  
+ .method getPreloadedDrawable(JI)Landroid/graphics/drawable/Drawable$ConstantState;
+     .locals 2
+     .parameter "key"
+     .parameter "id"
+ 
+     .prologue
+     sget-object v1, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v1, p1, p2}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
+ 
+     move-result-object v0
+ 
+     check-cast v0, Landroid/graphics/drawable/Drawable$ConstantState;
+ 
+     .local v0, cs:Landroid/graphics/drawable/Drawable$ConstantState;
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {p0, p3}, Landroid/content/res/Resources;->isPreloadOverlayed(I)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     const/4 v0, 0x0
+ 
+     :cond_0
+     return-object v0
+ .end method
+ 
  .method public getQuantityString(II)Ljava/lang/String;
      .locals 1
      .parameter "id"
***************
*** 2976,2981 ****
--- 3045,3069 ----
      return-object v0
  .end method
  
+ .method isPreloadOverlayed(I)Z
+     .locals 1
+     .parameter "id"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
+ .method isPreloading()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/content/res/Resources;->mPreloading:Z
+ 
+     return v0
+ .end method
+ 
  .method loadColorStateList(Landroid/util/TypedValue;I)Landroid/content/res/ColorStateList;
      .locals 12
      .parameter "value"
***************
*** 3303,3308 ****
--- 3391,3400 ----
      .locals 17
      .parameter "value"
      .parameter "id"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Landroid/content/res/Resources$NotFoundException;
***************
*** 3446,3460 ****
      .end local v4           #dr:Landroid/graphics/drawable/Drawable;
      .restart local v3       #dr:Landroid/graphics/drawable/Drawable;
      :cond_5
!     sget-object v13, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
! 
!     invoke-virtual {v13, v9, v10}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
  
!     move-result-object v13
  
!     check-cast v13, Landroid/graphics/drawable/Drawable$ConstantState;
  
!     move-object v2, v13
  
      goto :goto_2
  
--- 3538,3550 ----
      .end local v4           #dr:Landroid/graphics/drawable/Drawable;
      .restart local v3       #dr:Landroid/graphics/drawable/Drawable;
      :cond_5
!     move-object/from16 v0, p0
  
!     move/from16 v1, p2
  
!     invoke-virtual {v0, v9, v10, v1}, Landroid/content/res/Resources;->getPreloadedDrawable(JI)Landroid/graphics/drawable/Drawable$ConstantState;
  
!     move-result-object v2
  
      goto :goto_2
  
***************
*** 3618,3630 ****
      move-result-object v7
  
      .local v7, is:Ljava/io/InputStream;
      const/4 v13, 0x0
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, p1
  
!     invoke-static {v0, v1, v7, v6, v13}, Landroid/graphics/drawable/Drawable;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
--- 3708,3722 ----
      move-result-object v7
  
      .local v7, is:Ljava/io/InputStream;
+     invoke-static/range {p2 .. p2}, Landroid/content/res/Resources$Injector;->setDrawableId(I)V
+ 
      const/4 v13, 0x0
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, p1
  
!     invoke-static {v0, v1, v7, v6, v13}, Landroid/content/res/Resources$Injector;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
***************
*** 3741,3746 ****
--- 3833,3863 ----
      goto :goto_5
  .end method
  
+ .method loadOverlayDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
+     .locals 1
+     .parameter "value"
+     .parameter "id"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
+ 
+ .method loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
+     .locals 0
+     .parameter "array"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     return-object p1
+ .end method
+ 
  .method loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
      .locals 5
      .parameter "id"
***************
*** 4134,4140 ****
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     return-object v0
  .end method
  
  .method public obtainTypedArray(I)Landroid/content/res/TypedArray;
--- 4251,4261 ----
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     invoke-virtual {p0, v0}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v3
! 
!     return-object v3
  .end method
  
  .method public obtainTypedArray(I)Landroid/content/res/TypedArray;
***************
*** 4206,4212 ****
  
      aput v4, v2, v4
  
!     return-object v0
  .end method
  
  .method public openRawResource(I)Ljava/io/InputStream;
--- 4327,4337 ----
  
      aput v4, v2, v4
  
!     invoke-virtual {p0, v0}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v2
! 
!     return-object v2
  .end method
  
  .method public openRawResource(I)Ljava/io/InputStream;
***************
*** 4835,4840 ****
--- 4960,4968 ----
      .parameter "config"
      .parameter "metrics"
      .parameter "compat"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      move-object/from16 v0, p0
***************
*** 4946,4952 ****
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v20
  
      :cond_5
      move-object/from16 v0, p0
--- 5074,5086 ----
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v2
! 
!     const/high16 v3, -0x8000
! 
!     and-int v3, v3, v20
! 
!     or-int v20, v2, v3
  
      :cond_5
      move-object/from16 v0, p0
