*** ./smali/android/telephony/SmsMessage.smali	2015-04-19 13:46:57.893101977 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/telephony/SmsMessage.smali	2015-04-19 13:48:26.313105280 +0800
***************
*** 825,830 ****
--- 825,839 ----
      return-object v0
  .end method
  
+ .method public getMessageSimId()I
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
  .method public getOriginatingAddress()Ljava/lang/String;
      .locals 1
  
