*** ./smali/android/telephony/TelephonyManager.smali	2015-04-19 13:46:58.333101994 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/telephony/TelephonyManager.smali	2015-04-19 13:48:26.737105295 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "TelephonyManager.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/telephony/TelephonyManager$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field public static final ACTION_PHONE_STATE_CHANGED:Ljava/lang/String; = "android.intent.action.PHONE_STATE"
  
***************
*** 218,223 ****
--- 226,243 ----
      return-object v0
  .end method
  
+ .method static getContext()Landroid/content/Context;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-object v0, Landroid/telephony/TelephonyManager;->sContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public static getDefault()Landroid/telephony/TelephonyManager;
      .locals 1
  
***************
*** 1413,1418 ****
--- 1433,1441 ----
  
  .method public getSimOperator()Ljava/lang/String;
      .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const-string v0, "gsm.sim.operator.numeric"
***************
*** 1421,1426 ****
--- 1444,1453 ----
  
      move-result-object v0
  
+     invoke-static {v0}, Landroid/telephony/TelephonyManager$Injector;->getSimOperator(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
      return-object v0
  .end method
  
