*** ./smali/com/android/internal/app/LocalePicker.smali	2015-04-19 13:47:18.521102748 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/app/LocalePicker.smali	2015-04-19 13:48:46.141106020 +0800
***************
*** 64,69 ****
--- 64,73 ----
      .parameter "context"
      .parameter "layoutId"
      .parameter "fieldId"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
      .annotation system Ldalvik/annotation/Signature;
          value = {
              "(",
***************
*** 95,100 ****
--- 99,110 ----
      move-result-object v13
  
      .local v13, locales:[Ljava/lang/String;
+     move-object/from16 v0, v16
+ 
+     invoke-static {v0, v13}, Lmiui/util/ExtraLocalePicker;->filterLocale(Landroid/content/res/Resources;[Ljava/lang/String;)[Ljava/lang/String;
+ 
+     move-result-object v13
+ 
      const v20, 0x1070007
  
      move-object/from16 v0, v16
***************
*** 385,390 ****
--- 395,402 ----
      :cond_4
      invoke-static {v12}, Ljava/util/Arrays;->sort([Ljava/lang/Object;)V
  
+     invoke-static {v12}, Lmiui/util/ExtraLocalePicker;->adjustLocaleOrder([Lcom/android/internal/app/LocalePicker$LocaleInfo;)V
+ 
      new-instance v20, Landroid/widget/ArrayAdapter;
  
      move-object/from16 v0, v20
