*** ./smali/com/android/internal/app/ResolverActivity.smali	2015-04-19 13:47:18.249102738 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/app/ResolverActivity.smali	2015-04-19 13:48:45.861106010 +0800
***************
*** 11,17 ****
      value = {
          Lcom/android/internal/app/ResolverActivity$ItemLongClickListener;,
          Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;,
!         Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;
      }
  .end annotation
  
--- 11,18 ----
      value = {
          Lcom/android/internal/app/ResolverActivity$ItemLongClickListener;,
          Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;,
!         Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;,
!         Lcom/android/internal/app/ResolverActivity$Injector;
      }
  .end annotation
  
***************
*** 125,130 ****
--- 126,158 ----
      return v0
  .end method
  
+ .method private checkOption(Z)Z
+     .locals 1
+     .parameter "alwaysUseOption"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     if-eqz p1, :cond_0
+ 
+     invoke-static {p0}, Lmiui/util/UiUtils;->isV5Ui(Landroid/content/Context;)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
+     const/4 v0, 0x1
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method private makeMyIntent()Landroid/content/Intent;
      .locals 3
  
***************
*** 365,371 ****
  
      .prologue
      .local p5, rList:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
!     const v1, 0x1030302
  
      invoke-virtual {p0, v1}, Lcom/android/internal/app/ResolverActivity;->setTheme(I)V
  
--- 393,399 ----
  
      .prologue
      .local p5, rList:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
!     const v1, 0x60d003e
  
      invoke-virtual {p0, v1}, Lcom/android/internal/app/ResolverActivity;->setTheme(I)V
  
***************
*** 399,411 ****
  
      iput-boolean v0, p0, Lcom/android/internal/app/ResolverActivity;->mAlwaysUseOption:Z
  
!     invoke-virtual {p0}, Lcom/android/internal/app/ResolverActivity;->getResources()Landroid/content/res/Resources;
! 
!     move-result-object v1
! 
!     const v2, 0x10e0032
! 
!     invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getInteger(I)I
  
      move-result v1
  
--- 427,433 ----
  
      iput-boolean v0, p0, Lcom/android/internal/app/ResolverActivity;->mAlwaysUseOption:Z
  
!     invoke-static {p0}, Lcom/android/internal/app/ResolverActivity$Injector;->getMaxColumns(Lcom/android/internal/app/ResolverActivity;)I
  
      move-result v1
  
***************
*** 499,504 ****
--- 521,535 ----
  
      :cond_1
      :goto_1
+     iget-object v0, v9, Lcom/android/internal/app/AlertController$AlertParams;->mView:Landroid/view/View;
+ 
+     if-eqz v0, :cond_miui_00
+ 
+     move/from16 v0, p6
+ 
+     invoke-static {p0, v0}, Lcom/android/internal/app/ResolverActivity$Injector;->initialize(Lcom/android/internal/app/ResolverActivity;Z)V
+ 
+     :cond_miui_00
      return-void
  
      .end local v8           #am:Landroid/app/ActivityManager;
***************
*** 581,587 ****
      :goto_2
      invoke-virtual {p0}, Lcom/android/internal/app/ResolverActivity;->setupAlert()V
  
!     if-eqz p6, :cond_1
  
      const v1, 0x102026d
  
--- 612,624 ----
      :goto_2
      invoke-virtual {p0}, Lcom/android/internal/app/ResolverActivity;->setupAlert()V
  
!     move/from16 v0, p6
! 
!     invoke-direct {p0, v0}, Lcom/android/internal/app/ResolverActivity;->checkOption(Z)Z
! 
!     move-result v1
! 
!     if-eqz v1, :cond_1
  
      const v1, 0x102026d
  
***************
*** 1346,1351 ****
--- 1383,1401 ----
  
      return-void
  .end method
+ 
+ .method setAlwaysUseOption(Z)V
+     .locals 0
+     .parameter "alwaysUseOption"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/app/ResolverActivity;->mAlwaysUseOption:Z
+ 
+     return-void
+ .end method
  
  .method showAppDetails(Landroid/content/pm/ResolveInfo;)V
      .locals 5
