*** ./smali/com/android/internal/app/ResolverActivity$ResolveListAdapter.smali	2015-04-19 13:47:18.217102736 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/app/ResolverActivity$ResolveListAdapter.smali	2015-04-19 13:48:45.845106009 +0800
***************
*** 135,140 ****
--- 135,143 ----
      .locals 5
      .parameter "view"
      .parameter "info"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const v3, 0x1020014
***************
*** 194,200 ****
  
      iget-object v4, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->ri:Landroid/content/pm/ResolveInfo;
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/app/ResolverActivity;->loadIconForResolveInfo(Landroid/content/pm/ResolveInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
--- 197,203 ----
  
      iget-object v4, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->ri:Landroid/content/pm/ResolveInfo;
  
!     invoke-static {v3, v4}, Lcom/android/internal/app/ResolverActivity$Injector;->loadIconForResolveInfo(Lcom/android/internal/app/ResolverActivity;Landroid/content/pm/ResolveInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
***************
*** 205,210 ****
--- 208,223 ----
  
      invoke-virtual {v0, v3}, Landroid/widget/ImageView;->setImageDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     iget-object v3, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->this$0:Lcom/android/internal/app/ResolverActivity;
+ 
+     iget-object v4, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
+ 
+     invoke-interface {v4, p2}, Ljava/util/List;->indexOf(Ljava/lang/Object;)I
+ 
+     move-result v4
+ 
+     invoke-static {v0, v3, p0, v4}, Lcom/android/internal/app/ResolverActivity$Injector;->addListener(Landroid/view/View;Lcom/android/internal/app/ResolverActivity;Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;I)V
+ 
      return-void
  
      :cond_1
***************
*** 1121,1126 ****
--- 1134,1148 ----
      move-result-object v1
  
      .local v1, lp:Landroid/view/ViewGroup$LayoutParams;
+     iget v3, v1, Landroid/view/ViewGroup$LayoutParams;->width:I
+ 
+     if-eqz v3, :cond_miui_0
+ 
+     iget v3, v1, Landroid/view/ViewGroup$LayoutParams;->height:I
+ 
+     if-nez v3, :cond_miui_1
+ 
+     :cond_miui_0
      iget-object v3, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->this$0:Lcom/android/internal/app/ResolverActivity;
  
      #getter for: Lcom/android/internal/app/ResolverActivity;->mIconSize:I
***************
*** 1134,1139 ****
--- 1156,1162 ----
  
      .end local v0           #icon:Landroid/widget/ImageView;
      .end local v1           #lp:Landroid/view/ViewGroup$LayoutParams;
+     :cond_miui_1
      :goto_0
      iget-object v3, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
