*** ./smali/android/net/Proxy.smali	2015-04-19 13:46:53.649101819 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/net/Proxy.smali	2015-04-19 13:48:22.161105124 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;,
!         Landroid/net/Proxy$Injector;
      }
  .end annotation
  
***************
*** 257,262 ****
--- 258,266 ----
      .locals 6
      .parameter "context"
      .parameter "url"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-static {p0, p1}, Landroid/net/Proxy;->getProxy(Landroid/content/Context;Ljava/lang/String;)Ljava/net/Proxy;
***************
*** 287,293 ****
      .local v1, sa:Ljava/net/InetSocketAddress;
      new-instance v2, Lorg/apache/http/HttpHost;
  
!     invoke-virtual {v1}, Ljava/net/InetSocketAddress;->getHostName()Ljava/lang/String;
  
      move-result-object v3
  
--- 291,297 ----
      .local v1, sa:Ljava/net/InetSocketAddress;
      new-instance v2, Lorg/apache/http/HttpHost;
  
!     invoke-static {v1}, Landroid/net/Proxy$Injector;->getHostName(Ljava/net/InetSocketAddress;)Ljava/lang/String;
  
      move-result-object v3
  
