*** ./smali/com/android/internal/os/RuntimeInit.smali	2015-04-19 13:47:25.501103008 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/os/RuntimeInit.smali	2015-04-19 13:48:55.205106359 +0800
***************
*** 7,13 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/os/RuntimeInit$Arguments;,
!         Lcom/android/internal/os/RuntimeInit$UncaughtHandler;
      }
  .end annotation
  
--- 7,14 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/os/RuntimeInit$Arguments;,
!         Lcom/android/internal/os/RuntimeInit$UncaughtHandler;,
!         Lcom/android/internal/os/RuntimeInit$Injector;
      }
  .end annotation
  
***************
*** 137,142 ****
--- 138,157 ----
      goto :goto_0
  .end method
  
+ .method public static callGetDefaultUserAgent()Ljava/lang/String;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Lcom/android/internal/os/RuntimeInit;->getDefaultUserAgent()Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
  .method private static final commonInit()V
      .locals 4
  
***************
*** 220,225 ****
--- 235,243 ----
  
  .method private static getDefaultUserAgent()Ljava/lang/String;
      .locals 6
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 285,291 ****
  
      .end local v1           #model:Ljava/lang/String;
      :cond_0
!     sget-object v0, Landroid/os/Build;->ID:Ljava/lang/String;
  
      .local v0, id:Ljava/lang/String;
      invoke-virtual {v0}, Ljava/lang/String;->length()I
--- 303,313 ----
  
      .end local v1           #model:Ljava/lang/String;
      :cond_0
!     sget-object v4, Landroid/os/Build;->ID:Ljava/lang/String;
! 
!     invoke-static {v2, v4}, Lcom/android/internal/os/RuntimeInit$Injector;->appendMiuiVersion(Ljava/lang/StringBuilder;Ljava/lang/String;)Ljava/lang/String;
! 
!     move-result-object v0
  
      .local v0, id:Ljava/lang/String;
      invoke-virtual {v0}, Ljava/lang/String;->length()I
