*** ./smali/com/android/internal/os/ZygoteInit.smali	2015-04-19 13:47:25.309103001 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/os/ZygoteInit.smali	2015-04-19 13:48:55.017106352 +0800
***************
*** 419,424 ****
--- 419,426 ----
      :try_start_0
      invoke-static {}, Lcom/android/internal/os/SamplingProfilerIntegration;->start()V
  
+     invoke-static {}, Lmiui/security/SecurityManager;->init()V
+ 
      invoke-static {}, Lcom/android/internal/os/ZygoteInit;->registerZygoteSocket()V
  
      const/16 v2, 0xbcc
