*** ./smali/com/android/server/AppWidgetService$2.smali	2015-04-19 13:47:29.065103142 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/AppWidgetService$2.smali	2015-04-19 13:48:58.757106491 +0800
***************
*** 37,42 ****
--- 37,45 ----
      .locals 4
      .parameter "context"
      .parameter "intent"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
***************
*** 44,49 ****
--- 47,63 ----
      move-result-object v0
  
      .local v0, action:Ljava/lang/String;
+     iget-object v3, p0, Lcom/android/server/AppWidgetService$2;->this$0:Lcom/android/server/AppWidgetService;
+ 
+     invoke-static {v3, v0}, Lcom/android/server/AppWidgetService$Injector;->handleAction(Lcom/android/server/AppWidgetService;Ljava/lang/String;)Z
+ 
+     move-result v3
+ 
+     if-eqz v3, :cond_3
+ 
+     return-void
+ 
+     :cond_3
      const-string v3, "android.intent.action.BOOT_COMPLETED"
  
      invoke-virtual {v3, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
