*** ./smali/com/android/server/AttributeCache.smali	2015-04-19 13:47:36.133103405 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/AttributeCache.smali	2015-04-19 13:49:05.801106754 +0800
***************
*** 114,125 ****
      :try_start_0
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
!     invoke-virtual {v7, p1}, Ljava/util/WeakHashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v5
  
-     check-cast v5, Lcom/android/server/AttributeCache$Package;
- 
      .local v5, pkg:Lcom/android/server/AttributeCache$Package;
      const/4 v4, 0x0
  
--- 114,123 ----
      :try_start_0
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
!     invoke-static {v7, p1}, Lcom/android/server/AttributeCache$Injector;->getPackage(Ljava/util/WeakHashMap;Ljava/lang/String;)Lcom/android/server/AttributeCache$Package;
  
      move-result-object v5
  
      .local v5, pkg:Lcom/android/server/AttributeCache$Package;
      const/4 v4, 0x0
  
***************
*** 220,226 ****
      .restart local v5       #pkg:Lcom/android/server/AttributeCache$Package;
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
!     invoke-virtual {v7, p1, v5}, Ljava/util/WeakHashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      .end local v0           #context:Landroid/content/Context;
      :cond_2
--- 218,224 ----
      .restart local v5       #pkg:Lcom/android/server/AttributeCache$Package;
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
!     invoke-static {v7, p1, v5}, Lcom/android/server/AttributeCache$Injector;->putPackage(Ljava/util/WeakHashMap;Ljava/lang/String;Lcom/android/server/AttributeCache$Package;)V
  
      .end local v0           #context:Landroid/content/Context;
      :cond_2
