*** ./smali/com/android/server/ConnectivityService.smali	2015-04-19 13:47:34.501103345 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/ConnectivityService.smali	2015-04-19 13:49:04.069106690 +0800
***************
*** 10,16 ****
          Lcom/android/server/ConnectivityService$SettingsObserver;,
          Lcom/android/server/ConnectivityService$MyHandler;,
          Lcom/android/server/ConnectivityService$FeatureUser;,
!         Lcom/android/server/ConnectivityService$RadioAttributes;
      }
  .end annotation
  
--- 10,17 ----
          Lcom/android/server/ConnectivityService$SettingsObserver;,
          Lcom/android/server/ConnectivityService$MyHandler;,
          Lcom/android/server/ConnectivityService$FeatureUser;,
!         Lcom/android/server/ConnectivityService$RadioAttributes;,
!         Lcom/android/server/ConnectivityService$Injector;
      }
  .end annotation
  
***************
*** 201,206 ****
--- 202,210 ----
      .parameter "netd"
      .parameter "statsService"
      .parameter "policyManager"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      invoke-direct/range {p0 .. p0}, Landroid/net/IConnectivityManager$Stub;-><init>()V
***************
*** 343,348 ****
--- 347,354 ----
  
      invoke-direct {v0, v2}, Lcom/android/server/ConnectivityService;->log(Ljava/lang/String;)V
  
+     invoke-static/range {p1 .. p1}, Lcom/miui/server/FirewallService;->setupService(Landroid/content/Context;)V
+ 
      new-instance v13, Landroid/os/HandlerThread;
  
      const-string v2, "ConnectivityServiceThread"
***************
*** 6865,6870 ****
--- 6871,6878 ----
      move-result v8
  
      .local v8, usedNetworkType:I
+     invoke-static {p1, v8}, Lcom/android/server/ConnectivityService$Injector;->stopUsingNetworkFeature(Lcom/android/server/ConnectivityService$FeatureUser;I)V
+ 
      iget-object v10, p0, Lcom/android/server/ConnectivityService;->mNetTrackers:[Landroid/net/NetworkStateTracker;
  
      aget-object v6, v10, v8
***************
*** 10270,10275 ****
--- 10278,10286 ----
      .parameter "networkType"
      .parameter "feature"
      .parameter "binder"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const-wide/16 v14, 0x0
***************
*** 10823,10831 ****
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_2
  
      if-ltz v13, :cond_c
  
-     :try_start_8
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ConnectivityService;->mHandler:Landroid/os/Handler;
--- 10834,10844 ----
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_2
  
+     :try_start_8
+     invoke-static/range {v20 .. v20}, Lcom/android/server/ConnectivityService$Injector;->startUsingNetworkFeature(I)V
+ 
      if-ltz v13, :cond_c
  
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ConnectivityService;->mHandler:Landroid/os/Handler;
