*** ./smali/com/android/server/LightsService.smali	2015-04-19 13:47:31.117103218 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/LightsService.smali	2015-04-19 13:49:00.829106569 +0800
***************
*** 207,209 ****
--- 207,225 ----
  
      return-object v0
  .end method
+ 
+ .method setLight(ILcom/android/server/LightsService$Light;)V
+     .locals 1
+     .parameter "id"
+     .parameter "light"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/LightsService;->mLights:[Lcom/android/server/LightsService$Light;
+ 
+     aput-object p2, v0, p1
+ 
+     return-void
+ .end method
