*** ./smali/com/android/server/location/GpsLocationProvider.smali	2015-04-19 13:47:35.585103385 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/services.jar.out/./smali/com/android/server/location/GpsLocationProvider.smali	2015-04-19 13:49:05.189106731 +0800
***************
*** 11,17 ****
      value = {
          Lcom/android/server/location/GpsLocationProvider$GpsLocationProviderThread;,
          Lcom/android/server/location/GpsLocationProvider$ProviderHandler;,
!         Lcom/android/server/location/GpsLocationProvider$Listener;
      }
  .end annotation
  
--- 11,18 ----
      value = {
          Lcom/android/server/location/GpsLocationProvider$GpsLocationProviderThread;,
          Lcom/android/server/location/GpsLocationProvider$ProviderHandler;,
!         Lcom/android/server/location/GpsLocationProvider$Listener;,
!         Lcom/android/server/location/GpsLocationProvider$Injector;
      }
  .end annotation
  
***************
*** 3094,3099 ****
--- 3095,3103 ----
  .method private reportStatus(I)V
      .locals 10
      .parameter "status"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iget-object v8, p0, Lcom/android/server/location/GpsLocationProvider;->mListeners:Ljava/util/ArrayList;
***************
*** 3313,3318 ****
--- 3317,3324 ----
  
      invoke-virtual {v2, v7, v9}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     invoke-static {p0, v2}, Lcom/android/server/location/GpsLocationProvider$Injector;->appendUidExtra(Lcom/android/server/location/GpsLocationProvider;Landroid/content/Intent;)V
+ 
      iget-object v7, p0, Lcom/android/server/location/GpsLocationProvider;->mContext:Landroid/content/Context;
  
      invoke-virtual {v7, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
***************
*** 4333,4338 ****
--- 4339,4356 ----
      return v0
  .end method
  
+ .method getClientUids()Landroid/util/SparseIntArray;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider;->mClientUids:Landroid/util/SparseIntArray;
+ 
+     return-object v0
+ .end method
+ 
  .method public getGpsStatusProvider()Landroid/location/IGpsStatusProvider;
      .locals 1
  
***************
*** 4478,4483 ****
--- 4496,4513 ----
      return-object v0
  .end method
  
+ .method getNavigating()Z
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/server/location/GpsLocationProvider;->mNavigating:Z
+ 
+     return v0
+ .end method
+ 
  .method public getNetInitiatedListener()Landroid/location/INetInitiatedListener;
      .locals 1
  
