*** ./smali/android/webkit/WebSettings.smali	2015-04-19 13:47:12.017102505 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/webkit/WebSettings.smali	2015-04-19 13:48:39.713105780 +0800
***************
*** 78,83 ****
--- 78,106 ----
  .method public abstract getAllowUniversalAccessFromFileURLs()Z
  .end method
  
+ .method public declared-synchronized getBlockFloatPopupWindows()Z
+     .locals 1
+ 
+     .prologue
+     monitor-enter p0
+ 
+     :try_start_0
+     new-instance v0, Landroid/webkit/MustOverrideException;
+ 
+     invoke-direct {v0}, Landroid/webkit/MustOverrideException;-><init>()V
+ 
+     throw v0
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     :catchall_0
+     move-exception v0
+ 
+     monitor-exit p0
+ 
+     throw v0
+ .end method
+ 
  .method public declared-synchronized getBlockNetworkImage()Z
      .locals 1
  
***************
*** 941,946 ****
--- 964,993 ----
  
      .prologue
      monitor-enter p0
+ 
+     :try_start_0
+     new-instance v0, Landroid/webkit/MustOverrideException;
+ 
+     invoke-direct {v0}, Landroid/webkit/MustOverrideException;-><init>()V
+ 
+     throw v0
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     :catchall_0
+     move-exception v0
+ 
+     monitor-exit p0
+ 
+     throw v0
+ .end method
+ 
+ .method public declared-synchronized setBlockFloatPopupWindows(Z)V
+     .locals 1
+     .parameter "flag"
+ 
+     .prologue
+     monitor-enter p0
  
      :try_start_0
      new-instance v0, Landroid/webkit/MustOverrideException;
