*** ./smali/android/webkit/WebView.smali	2015-04-19 13:47:12.053102506 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/webkit/WebView.smali	2015-04-19 13:48:39.749105781 +0800
***************
*** 117,122 ****
--- 117,124 ----
  
      invoke-interface {v0, p4, p5}, Landroid/webkit/WebViewProvider;->init(Ljava/util/Map;Z)V
  
+     invoke-static {p1}, Lmiui/content/res/ThemeFontChangeHelper;->markWebViewCreated(Landroid/content/Context;)V
+ 
      return-void
  .end method
  
