*** ./smali/android/webkit/WebViewClassic.smali	2015-04-19 13:47:13.469102559 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/webkit/WebViewClassic.smali	2015-04-19 13:48:41.125105833 +0800
***************
*** 15811,15816 ****
--- 15811,15825 ----
      goto :goto_0
  .end method
  
+ .method public isHitSelectionHandle()Z
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
  .method public isPaused()Z
      .locals 1
  
***************
*** 15963,15968 ****
--- 15972,15986 ----
      goto :goto_0
  .end method
  
+ .method public isSelectingText()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/webkit/WebViewClassic;->mSelectingText:Z
+ 
+     return v0
+ .end method
+ 
  .method public loadData(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
      .locals 0
      .parameter "data"
***************
*** 20952,20958 ****
      goto :goto_0
  .end method
  
! .method sendOurVisibleRect()Landroid/graphics/Rect;
      .locals 4
  
      .prologue
--- 20970,20976 ----
      goto :goto_0
  .end method
  
! .method public sendOurVisibleRect()Landroid/graphics/Rect;
      .locals 4
  
      .prologue
***************
*** 22066,22071 ****
--- 22084,22120 ----
      return-void
  .end method
  
+ .method public setProxy(Ljava/lang/String;I)V
+     .locals 3
+     .parameter "host"
+     .parameter "port"
+ 
+     .prologue
+     const/4 v2, 0x0
+ 
+     const/16 v1, 0xc1
+ 
+     if-eqz p1, :cond_0
+ 
+     if-nez p2, :cond_1
+ 
+     :cond_0
+     invoke-static {v1, v2}, Landroid/webkit/WebViewCore;->sendStaticMessage(ILjava/lang/Object;)V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_1
+     new-instance v0, Landroid/net/ProxyProperties;
+ 
+     invoke-direct {v0, p1, p2, v2}, Landroid/net/ProxyProperties;-><init>(Ljava/lang/String;ILjava/lang/String;)V
+ 
+     .local v0, pp:Landroid/net/ProxyProperties;
+     invoke-static {v1, v0}, Landroid/webkit/WebViewCore;->sendStaticMessage(ILjava/lang/Object;)V
+ 
+     goto :goto_0
+ .end method
+ 
  .method public setScrollBarStyle(I)V
      .locals 1
      .parameter "style"
***************
*** 22898,22900 ****
--- 22947,22967 ----
  
      return v0
  .end method
+ 
+ .method public hideFloatView()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public checkIfSaveImageFromCacheAvailable(Ljava/lang/String;Ljava/lang/String;)Z
+     .locals 1
+     .parameter "imageUrl"
+     .parameter "path"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
