*** ./smali/android/net/wifi/WifiInfo.smali	2015-04-19 13:46:50.765101711 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiInfo.smali	2015-04-19 13:48:19.325105019 +0800
***************
*** 54,59 ****
--- 54,61 ----
  
  .field private mSupplicantState:Landroid/net/wifi/SupplicantState;
  
+ .field private mVendorInfo:Ljava/lang/String;
+ 
  
  # direct methods
  .method static constructor <clinit>()V
***************
*** 263,268 ****
--- 265,274 ----
  
      iput-boolean v0, p0, Landroid/net/wifi/WifiInfo;->mMeteredHint:Z
  
+     iget-object v0, p1, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     iput-object v0, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
      :cond_0
      return-void
  .end method
***************
*** 535,540 ****
--- 541,555 ----
      return-object v0
  .end method
  
+ .method public getVendorInfo()Ljava/lang/String;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     return-object v0
+ .end method
+ 
  .method public getSupplicantState()Landroid/net/wifi/SupplicantState;
      .locals 1
  
***************
*** 662,667 ****
--- 677,692 ----
      return-void
  .end method
  
+ .method public setVendorInfo(Ljava/lang/String;)V
+     .locals 0
+     .parameter "vendorInfo"
+ 
+     .prologue
+     iput-object p1, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     return-void
+ .end method
+ 
  .method public toString()Ljava/lang/String;
      .locals 4
  
***************
*** 873,878 ****
--- 898,907 ----
      :goto_1
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     iget-object v0, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
      iget-object v0, p0, Landroid/net/wifi/WifiInfo;->mSupplicantState:Landroid/net/wifi/SupplicantState;
  
      invoke-virtual {v0, p1, p2}, Landroid/net/wifi/SupplicantState;->writeToParcel(Landroid/os/Parcel;I)V
***************
*** 889,891 ****
--- 918,931 ----
  
      goto :goto_1
  .end method
+ 
+ .method static synthetic access_setVendorInfo(Landroid/net/wifi/WifiInfo;Ljava/lang/String;)Ljava/lang/String;
+     .locals 0
+     .parameter "x0"
+     .parameter "x1"
+ 
+     .prologue
+     iput-object p1, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     return-object p1
+ .end method
