*** ./smali/android/net/wifi/WifiConfiguration.smali	2015-04-19 13:46:50.717101709 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiConfiguration.smali	2015-04-19 13:48:19.269105016 +0800
***************
*** 120,129 ****
--- 120,159 ----
  
  .field public priority:I
  
+ .field public private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field public proxySettings:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
  .field public status:I
  
+ .field public wapiASCert:Ljava/lang/String;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
+ .field public wapiPsk:Ljava/lang/String;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
+ .field public wapiPskType:I
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
+ .field public wapiUserCert:Ljava/lang/String;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field public wepKeys:[Ljava/lang/String;
  
  .field public wepTxKeyIndex:I
***************
*** 187,192 ****
--- 217,230 ----
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     const-string v6, "private_key"
+ 
+     invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
+ 
+     iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
+ 
+     new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
+ 
      const-string v6, "eap"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
***************
*** 447,452 ****
--- 485,498 ----
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     const-string v2, "private_key"
+ 
+     invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
+ 
+     iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
+ 
+     new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
+ 
      const-string v2, "eap"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
