*** ./smali/android/net/wifi/WifiInfo$1.smali	2015-04-19 13:46:50.617101706 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiInfo$1.smali	2015-04-19 13:48:19.201105014 +0800
***************
*** 120,125 ****
--- 120,131 ----
      #setter for: Landroid/net/wifi/WifiInfo;->mMeteredHint:Z
      invoke-static {v0, v1}, Landroid/net/wifi/WifiInfo;->access$202(Landroid/net/wifi/WifiInfo;Z)Z
  
+     invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-static {v0, v1}, Landroid/net/wifi/WifiInfo;->access_setVendorInfo(Landroid/net/wifi/WifiInfo;Ljava/lang/String;)Ljava/lang/String;
+ 
      sget-object v1, Landroid/net/wifi/SupplicantState;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v1, p1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
