*** ./smali/android/graphics/drawable/Drawable.smali	2015-04-19 13:47:09.597102414 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/graphics/drawable/Drawable.smali	2015-04-19 13:48:37.297105690 +0800
***************
*** 33,38 ****
--- 33,44 ----
  
  .field private mChangingConfigurations:I
  
+ .field mId:I
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field private mLevel:I
  
  .field private mStateSet:[I
***************
*** 82,87 ****
--- 88,97 ----
  
      iput-boolean v0, p0, Landroid/graphics/drawable/Drawable;->mVisible:Z
  
+     const/4 v0, -0x1
+ 
+     iput v0, p0, Landroid/graphics/drawable/Drawable;->mId:I
+ 
      return-void
  .end method
  
***************
*** 244,249 ****
--- 254,311 ----
      goto :goto_0
  .end method
  
+ .method public static createFromResourceStreamAndCloseStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
+     .locals 2
+     .parameter "res"
+     .parameter "value"
+     .parameter "is"
+     .parameter "srcName"
+     .parameter "opts"
+ 
+     .prologue
+     :try_start_0
+     invoke-static {p0, p1, p2, p3, p4}, Landroid/graphics/drawable/Drawable;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     move-result-object v0
+ 
+     if-eqz p2, :cond_0
+ 
+     :try_start_1
+     invoke-virtual {p2}, Ljava/io/InputStream;->close()V
+     :try_end_1
+     .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
+ 
+     :cond_0
+     :goto_0
+     return-object v0
+ 
+     :catchall_0
+     move-exception v0
+ 
+     if-eqz p2, :cond_1
+ 
+     :try_start_2
+     invoke-virtual {p2}, Ljava/io/InputStream;->close()V
+     :try_end_2
+     .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_1
+ 
+     :cond_1
+     :goto_1
+     throw v0
+ 
+     :catch_0
+     move-exception v1
+ 
+     goto :goto_0
+ 
+     :catch_1
+     move-exception v1
+ 
+     goto :goto_1
+ .end method
+ 
  .method public static createFromStream(Ljava/io/InputStream;Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
      .locals 1
      .parameter "is"
***************
*** 863,868 ****
--- 925,942 ----
      return-object p0
  .end method
  
+ .method public getId()I
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget v0, p0, Landroid/graphics/drawable/Drawable;->mId:I
+ 
+     return v0
+ .end method
+ 
  .method public getIntrinsicHeight()I
      .locals 1
  
***************
*** 1292,1297 ****
--- 1366,1384 ----
      return-void
  .end method
  
+ .method public setId(I)V
+     .locals 0
+     .parameter "id"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput p1, p0, Landroid/graphics/drawable/Drawable;->mId:I
+ 
+     return-void
+ .end method
+ 
  .method public final setLevel(I)Z
      .locals 1
      .parameter "level"
