*** ./smali/com/android/internal/view/ActionBarPolicy.smali	2015-04-19 13:47:15.973102653 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/view/ActionBarPolicy.smali	2015-04-19 13:48:43.581105924 +0800
***************
*** 227,232 ****
--- 227,236 ----
      .locals 1
  
      .prologue
+     const/4 v0, 0x1
+     
+     return v0
+     
      iget-object v0, p0, Lcom/android/internal/view/ActionBarPolicy;->mContext:Landroid/content/Context;
  
      invoke-static {v0}, Landroid/view/ViewConfiguration;->get(Landroid/content/Context;)Landroid/view/ViewConfiguration;
