*** ./smali/android/text/format/DateFormat.smali	2015-04-19 13:46:35.361101136 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/android/text/format/DateFormat.smali	2015-04-19 13:48:00.325104309 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "DateFormat.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/text/format/DateFormat$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field public static final AM_PM:C = 'a'
  
***************
*** 1140,1145 ****
--- 1148,1156 ----
  .method public static is24HourFormat(Landroid/content/Context;)Z
      .locals 9
      .parameter "context"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      const/4 v6, 0x0
***************
*** 1157,1162 ****
--- 1168,1177 ----
      move-result-object v5
  
      .local v5, value:Ljava/lang/String;
+     invoke-static {p0, v5}, Landroid/text/format/DateFormat$Injector;->check24HourFormatForChina(Landroid/content/Context;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v5
+ 
      if-nez v5, :cond_1
  
      invoke-virtual {p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
