*** ./smali/com/android/internal/telephony/DataConnectionTracker.smali	2015-04-19 13:47:25.197102997 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/DataConnectionTracker.smali	2015-04-19 13:48:54.921106348 +0800
***************
*** 10,16 ****
          Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;,
          Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;,
          Lcom/android/internal/telephony/DataConnectionTracker$Activity;,
!         Lcom/android/internal/telephony/DataConnectionTracker$State;
      }
  .end annotation
  
--- 10,17 ----
          Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;,
          Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;,
          Lcom/android/internal/telephony/DataConnectionTracker$Activity;,
!         Lcom/android/internal/telephony/DataConnectionTracker$State;,
!         Lcom/android/internal/telephony/DataConnectionTracker$Injector;
      }
  .end annotation
  
***************
*** 2456,2461 ****
--- 2457,2465 ----
  
  .method public getAnyDataEnabled()Z
      .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
***************
*** 2483,2488 ****
--- 2487,2496 ----
  
      .local v0, result:Z
      :goto_0
+     invoke-static {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker$Injector;->getAnyDataEnabled(Lcom/android/internal/telephony/DataConnectionTracker;Z)Z
+ 
+     move-result v0
+ 
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 3469,3474 ****
--- 3477,3528 ----
      throw v1
  .end method
  
+ .method protected isMmsDataEnabled()Z
+     .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
+ 
+     const-string v2, "mms"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v1
+ 
+     const-string v2, "always_enable_mms"
+ 
+     invoke-static {v1, v2, v0}, Landroid/provider/Settings$System;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
+ 
+     move-result v1
+ 
+     if-ne v1, v0, :cond_0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method protected abstract log(Ljava/lang/String;)V
  .end method
  
