*** ./smali/com/android/internal/telephony/PhoneBase.smali	2015-04-19 13:47:22.833102909 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/PhoneBase.smali	2015-04-19 13:48:52.389106253 +0800
***************
*** 3515,3517 ****
--- 3515,3575 ----
  
      return-void
  .end method
+ 
+ .method protected setDeviceIdSystemProperty()V
+     .locals 2
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/telephony/PhoneBase;->getDeviceId()Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
+     const-string v0, "ro.ril.miui.imei"
+ 
+     invoke-virtual {p0}, Lcom/android/internal/telephony/PhoneBase;->getDeviceId()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/PhoneBase;->setSystemProperty(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     :cond_0
+     return-void
+ .end method
+ 
+ .method protected sendDeviceIdReadyBroadcast()V
+     .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     new-instance v0, Landroid/content/Intent;
+ 
+     const-string v1, "android.intent.action.DEVICE_ID_READY"
+ 
+     invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
+ 
+     .local v0, intent:Landroid/content/Intent;
+     const-string v1, "device_id"
+ 
+     invoke-virtual {p0}, Lcom/android/internal/telephony/PhoneBase;->getDeviceId()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/PhoneBase;->mContext:Landroid/content/Context;
+ 
+     const-string v2, "android.permission.READ_PHONE_STATE"
+ 
+     invoke-virtual {v1, v0, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
+ 
+     return-void
+ .end method
+ 
