*** ./smali/com/android/internal/telephony/SMSDispatcher.smali	2015-04-19 13:47:22.893102911 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/SMSDispatcher.smali	2015-04-19 13:48:52.465106256 +0800
***************
*** 7,13 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/telephony/SMSDispatcher$ConfirmDialogListener;,
!         Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      }
  .end annotation
  
--- 7,14 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/telephony/SMSDispatcher$ConfirmDialogListener;,
!         Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;,
!         Lcom/android/internal/telephony/SMSDispatcher$Injector;
      }
  .end annotation
  
***************
*** 897,903 ****
  
      move-result-object v1
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
  
      move-result v8
  
--- 898,908 ----
  
      move-result-object v1
  
!     invoke-virtual {p1}, Lcom/android/internal/telephony/SmsMessageBase;->getOriginatingAddress()Ljava/lang/String;
! 
!     move-result-object v2
! 
!     invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([BLjava/lang/String;)I
  
      move-result v8
  
***************
*** 2165,2171 ****
  
      iget-object v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
  
!     invoke-virtual {v3, v13}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
  
      move-result v3
  
--- 2170,2178 ----
  
      iget-object v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
  
!     move-object/from16 v0, p2
! 
!     invoke-virtual {v3, v13, v0}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([BLjava/lang/String;)I
  
      move-result v3
  
***************
*** 2260,2266 ****
  
      move-result-object v4
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
  
      move-result v3
  
--- 2267,2275 ----
  
      move-result-object v4
  
!     move-object/from16 v0, p2
! 
!     invoke-virtual {v3, v4, v0}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([BLjava/lang/String;)I
  
      move-result v3
  
