*** ./smali/com/android/internal/telephony/SmsUsageMonitor.smali	2015-04-19 13:47:20.193102810 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/SmsUsageMonitor.smali	2015-04-19 13:48:47.861106084 +0800
***************
*** 300,307 ****
--- 300,323 ----
      .locals 3
      .parameter "appName"
      .parameter "smsWaiting"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
  
      .prologue
+     const-string v1, "com.android.mms"
+ 
+     invoke-virtual {v1, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_ff
+ 
+     const/4 v1, 0x1
+ 
+     return v1
+ 
+     :cond_ff
      iget-object v2, p0, Lcom/android/internal/telephony/SmsUsageMonitor;->mSmsStamp:Ljava/util/HashMap;
  
      monitor-enter v2
