*** ./smali/com/android/internal/telephony/WapPushOverSms.smali	2015-04-19 13:47:22.877102910 +0800
--- /home/art-chen/MIUI/v5/s7568/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/WapPushOverSms.smali	2015-04-19 13:48:52.449106256 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/WapPushOverSms$WapPushConnection;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/WapPushOverSms$WapPushConnection;,
!         Lcom/android/internal/telephony/WapPushOverSms$Injector;
      }
  .end annotation
  
***************
*** 20,25 ****
--- 21,32 ----
  
  .field private final WAKE_LOCK_TIMEOUT:I
  
+ .field mAddress:Ljava/lang/String;
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  .field private final mContext:Landroid/content/Context;
  
  .field private mSmsDispatcher:Lcom/android/internal/telephony/SMSDispatcher;
***************
*** 599,604 ****
--- 606,625 ----
  
      invoke-virtual {v12, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     const-string v22, "address"
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v0, v0, Lcom/android/internal/telephony/WapPushOverSms;->mAddress:Ljava/lang/String;
+ 
+     move-object/from16 v23, v0
+ 
+     move-object/from16 v0, v22
+ 
+     move-object/from16 v1, v23
+ 
+     invoke-virtual {v12, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
+ 
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/telephony/WapPushOverSms;->mSmsDispatcher:Lcom/android/internal/telephony/SMSDispatcher;
***************
*** 623,625 ****
--- 644,688 ----
      .restart local v16       #permission:Ljava/lang/String;
      goto :goto_4
  .end method
+ 
+ .method public dispatchWapPdu([BLjava/lang/String;)I
+     .locals 1
+     .parameter "pdu"
+     .parameter "address"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-object p2, p0, Lcom/android/internal/telephony/WapPushOverSms;->mAddress:Ljava/lang/String;
+ 
+     invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
+ .method getContext()Landroid/content/Context;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/WapPushOverSms;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
+ .method getSmsDispatcher()Lcom/android/internal/telephony/SMSDispatcher;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/WapPushOverSms;->mSmsDispatcher:Lcom/android/internal/telephony/SMSDispatcher;
+ 
+     return-object v0
+ .end method
